Dataset,System,Bug ID,Creation Date,Title,Description,Ground Truth
FILE,DATAMONGO,DATAMONGO-398,2012-02-10T07:11:53.000-06:00,store org.osgi.service.event.Event,"When I attempt to save an object from org.apache.felix's versions of org.osgi.service.event.Event, the stack trace below occurs.
This appears to be due to the system attempting to save the topic field in Event as event.topics.
I am using the following dependency versions
cause by java.lang.IllegalArgumentException store in db
store in db
store in db","org.springframework.data.mongodb.core.convert.MappingMongoConverter
org.springframework.data.mongodb.core.convert.MappingMongoConverterUnitTests"
FILE,DATAMONGO,DATAMONGO-523,2012-09-01T03:39:51.000-05:00,@TypeAlias annotation not used with AbstractMongoConfiguration,"When using the AbstractMongoConfiguration without any further modifications regarding the converter (afterMappingMongoConverterCreation) the @TypeAlias annotation is not used when writing the _class property.
Seems like it always uses the SimpleTypeInformationMapper.
annotate @Document classes with @TypeAlias annotations",org.springframework.data.mongodb.core.convert.MappingMongoConverterUnitTests
FILE,DATAMONGO,DATAMONGO-629,2013-03-22T04:08:25.000-05:00,Different results when using count and find with the same criteria with 'id' field,"Assume we have following query:
mongoTemplate.find(q,java.util.HashMap.class,'test') gives following query (peeked in mongo console):
The same query, when used in count (mongoTemplate.count(q,'test')) gives:
This is inconsistent since we could have records with field id and they will be retrieved properly from the db.
Count on the other hand will give bad results since it uses _id field.
treat id _ id as same field treat id _ id in org.springframework.data.mongodb.core.convert.QueryMapper
This is probably the cause of the strange behaviour because find() method does not use QueryMapper and count() does.","org.springframework.data.mongodb.core.mapping.MongoMappingContext
org.springframework.data.mongodb.core.MongoTemplateTests
org.springframework.data.mongodb.core.convert.QueryMapper
org.springframework.data.mongodb.core.convert.QueryMapperUnitTests"
FILE,DATAMONGO,DATAMONGO-571,2012-11-09T08:00:10.000-06:00,Spring Data for MongoDb doesn't save null values when @Version is added to domain class,"1. Domain class is loaded from mongodb using CrudRepository.findOne() method.
2. The loaded instances any non id nor @Version annotated field is set to null.
3. The loaded instance is saved to same mongodb using CrudRepository.save() method.
4. The field that has been set to null doesnt write to database, its unchanged.
not use @Version annotation in domain class definition","org.springframework.data.mongodb.core.MongoTemplateTests
org.springframework.data.mongodb.core.query.Update"
FILE,DATAMONGO,DATAMONGO-688,2013-06-03T14:11:53.000-05:00,"There is no precedence between @Id annotation and field named ""id"" - both are attempted to be used","If you have a domain class with an attribute named ""id"", even if you annotate a DIFFERENT field with ""@Id"", spring-data-mongo errors trying to map 2 things to the id.
While the error is correct in that you can't have 2, it surprises me that one doesn't take precedence over the other.
find issue list in jira read from post read after hitting","org.springframework.data.mongodb.core.mapping.MongoPersistentProperty
org.springframework.data.mongodb.core.mapping.BasicMongoPersistentProperty
org.springframework.data.mongodb.core.mapping.BasicMongoPersistentEntity
org.springframework.data.mongodb.core.mapping.MongoMappingContextUnitTests"
FILE,DATAMONGO,DATAMONGO-392,2012-02-07T04:28:15.000-06:00,Updating an object does not write type information for objects to be updated,"I'm using quite complex domain model, that consist of instantiable domain classes as well as of abstract ones.
I used 1.0.0.
M5 version, and the type information (under _class key) was stored with object when it was necessary to be able to read it from database later.
work till upgrade work to 1.0.0
RELEASE version that broke my application as it saves the objects without type information and later it is impossible to read it back to java model.
What I found is that MappingMongoConverter.writeInternal(...) method that in turn calls addCustomTypeIfNecessary(...) (line 330) which puts type information into DBObject.
During execution of convertToMongoType(...) (at line 851) removeTypeInfoRecursively(...) is called which clears type data saved earlier under _class key.
comment out call
save type information to DBObject","org.springframework.data.mongodb.core.MongoTemplateTests
org.springframework.data.mongodb.core.MongoTemplate"
FILE,DATAMONGO,DATAMONGO-717,2013-07-10T11:13:46.000-05:00,Application context is not properly distributed to persistent entities,"The MongoMappingContext does not properly distribute the application context when set to persistent entities that have already been added.
Current code:
protect <T> BasicMongoPersistentEntity<T>
throw BeansException {
throw BeansException {
send application context to ALL create after point
see referenced URL poc
call initialize() on MongoMappingContext return object
throw ClassNotFoundException {",org.springframework.data.mongodb.config.AbstractMongoConfigurationUnitTests
FILE,DATAMONGO,DATAMONGO-721,2013-07-11T11:36:06.000-05:00,Polymorphic attribute type not persisted on update operations,"find with Spring data find for Mongo DB
Here is our situation: we have an entity which have an attribute which is a list of another kind of entity, like the code below.
annotate with @Document
When using MongoTemplate class with code such as below, the _class attribute is not inserted on the embedded document, so, if one of the items of the list attribute is a subclass of ChildClass, and ChildClass is an abstract class, we begin to face instantiation problems.
Here is one example of usage of MongoTemplate in which we found a problem.
If child is a subclass of ChildClass, the _class attribute is not added to the embedded document.",org.springframework.data.mongodb.core.convert.QueryMapper
FILE,DATAMONGO,DATAMONGO-602,2013-01-30T02:22:53.000-06:00,Querying with $in operator on the id field of type BigInteger returns zero results,"There is a problem when trying to query for documents with id field is in a given list of BigInteger values.
map id field as BigInteger
For example, these lines will work only if given id list contains only one item.
The query looks like this for a profileIds with multiple elements
and it looks like this when there is only one item in the list.
In the first case, query will return no results and in the second it will work and return an Iterable with one item.
be with representation
use decimal format not work with MongoDB",org.springframework.data.mongodb.core.MongoTemplateTests
FILE,DATAMONGO,DATAMONGO-805,2013-12-02T06:34:36.000-06:00,Excluding DBRef field in a query causes a MappingException,"Excluding a field in a query where the field is a DBRef as below throws a MappingException.
Exception trace:
find for class java.lang.Integer
I've attached a simple test case that throws the MappingException.
include for other fields","org.springframework.data.mongodb.core.convert.QueryMapper
org.springframework.data.mongodb.core.mapping.MappingTests
org.springframework.data.mongodb.core.convert.QueryMapperUnitTests"
FILE,DATAMONGO,DATAMONGO-897,2014-04-01T04:38:51.000-05:00,use @DbRef as target use interface as target,"NullPointerException is thrown when using MongoTemplate.findAndModify(...) with @DbRef and interface as @DbRef target.
attach project for more details
pass with Spring Commons 1.6.3","org.springframework.data.mongodb.core.convert.UpdateMapperUnitTests
org.springframework.data.mongodb.core.convert.QueryMapper"
FILE,DATAMONGO,DATAMONGO-647,2013-04-09T17:29:02.000-05:00,"Using ""OrderBy"" in ""query by method name"" ignores the @Field annotation for field alias.","I created a method using the ""query by method name"" approach:
Inside my Answer object, I have a field called ""Score"" that is annotated with
int score
When the query is run, the database attempts to sort the results by ""score"" rather than my ""sr"" field name.",org.springframework.data.mongodb.core.convert.QueryMapperUnitTests
FILE,DATAMONGO,DATAMONGO-938,2014-05-21T06:09:48.000-05:00,Exception when creating geo within Criteria using MapReduce,"I am getting an IllegalArgumentException when I try to query a MongoDB collection using a Criteria.within and a Box.
The exception reads:
not serialize class org.springframework.data.mongodb.core.query.GeoCommand","org.springframework.data.mongodb.core.mapreduce.MapReduceTests
org.springframework.data.mongodb.core.MongoTemplate"
FILE,DATAMONGO,DATAMONGO-987,2014-07-14T12:01:52.000-05:00,Problem with lazy loading in @DBRef when getting data using MongoTemplate,"The situation is simple: if we reference on an entity class another entity (both annotated with @Document) called Parent and Child.
Here is the code:
and Child class
// Than we try to load this same entity from the database
assertNull(persisted.getChild()); // it fails
bring lot of problems persist same entity by accident
I attached a project with the JUnit test which reproduces the problem for you.","org.springframework.data.mongodb.core.convert.MappingMongoConverter
org.springframework.data.mongodb.core.convert.DbRefMappingMongoConverterUnitTests"
FILE,DATAMONGO,DATAMONGO-1027,2014-08-16T13:49:22.000-05:00,Collection inherits complex index from embedded class/object,"Say I have two collections:
Books and Author
Author has a unique compound index and Book has a list of authors as property.
Now if I add different book with the same author to books, the insert is ignored because books inherits the index from authors.
On a subsequent context restart I'm also getting a duplicate key exception:
dup key","org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexResolver
org.springframework.data.mongodb.core.index.Indexed
org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexResolverUnitTests
org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexCreatorUnitTests
org.springframework.data.mongodb.core.index.CompoundIndex
org.springframework.data.mongodb.core.index.GeoSpatialIndexed"
FILE,DATAMONGO,DATAMONGO-1088,2014-11-07T03:08:58.000-06:00,"@Query $in does not remove ""_class"" property on collection of embedded objects","Following method on repository
generates incorrect query.
I attached test project demonstrating this bug.
relate bug to https://jira.spring.io/browse/DATAMONGO-893","org.springframework.data.mongodb.core.convert.MappingMongoConverter
org.springframework.data.mongodb.core.MongoTemplate"
FILE,DATAMONGO,DATAMONGO-1123,2014-12-17T09:39:36.000-06:00,"geoNear, does not return all matching elements, it returns only a max of 100 documents","I have the following query:
document geoNear method
return { @link georesults } for matching match given { @link nearquery }
I expect 1000 ""matching"" documents But i only get 100.
There is some default being set, that restricts the result to 100.
state in method
have pageable",org.springframework.data.mongodb.core.MongoOperations
FILE,DATAMONGO,DATAMONGO-1126,2014-12-21T06:03:21.000-06:00,keyword query findByInId with pageable,"I've been trying to use the In-keyword with identifiers and making the query pageable.
The query returns results but getTotalElements() and getTotalPages() always returns 0.
Also when you try to get any other page than 0, no results return.
use with member
Below is a strip down example I used for testing;
I've created 3 types and 10 items per those types, results in a total of 30 items.
Assert.assertEquals(30, results.getTotalElements()); // this is returning 0
Assert.assertEquals(6, results.getTotalPages());     // this is returning 0","org.springframework.data.mongodb.repository.Person
org.springframework.data.mongodb.repository.query.AbstractMongoQueryUnitTests
org.springframework.data.mongodb.core.MongoOperations
org.springframework.data.mongodb.core.MongoTemplate
org.springframework.data.mongodb.repository.query.AbstractMongoQuery"
FILE,DATAMONGO,DATAMONGO-1250,2015-07-03T21:07:44.000-05:00,Custom converter implementation not used in updates,"use in mongoTemplate.update*
I have a custom (de)serialiser for an enumerated type, and it works perfectly when saving and loading a @Document annotated POJO.
It also works when building and executing a Query object.
However when used in an Update, it is either ignored, or called in situations where it shouldn't.
Please clone https://github.com/patrickherrera/converter_test.git for a full test application.
In brief there is a POJO and for the purposes of the test it has a static enum with the desired converters:
return new converter <allocation, string> {
return new converter <string, allocation> {
throw new illegalargumentexception
use full Enum name use short code
register in Spring boot application entry point
return new customconversions
There is a unit test that drives a few scenarios:
save first allocation confirm conversion in query
confirm conversion in query save with valid allocation
try {
convert allocation not serialize class converter_test
check allocation update
update value get MyPersistantObject.Allocation
use MyPersistantObject.Allocation converter to string
check value update
make sense
save for object query for object call converters on document
store name store Enum code
By use of a positive and negative case, it appears that the converter is being called correctly when used in the Query builder.
When it comes to an Update, the Enum is unable to be serialised correctly, and an exception is thrown to that effect.
use code back by Querying back from DB
So it appears that the customer converter for converting from my Enum is not called in this situation.
update other value in document
The BigDecimal is converted to a String by an existing converter I assume, but then my customer converter is called to try and convert the numeric String into an Allocation Enum which of course fails.
debug code be in customconversions
That second variant never seems to be called in MappingMongoConverter, but perhaps if that type information was passed then it would not use my Allocation converter.
Without type information it seems the default is just to use the first converter that can handle the input type, in this case a String.
not handle custom one
not find workaround do update resort to Mongo","org.springframework.data.mongodb.core.convert.UpdateMapperUnitTests
org.springframework.data.mongodb.core.convert.UpdateMapper"
FILE,DATAMONGO,DATAMONGO-1263,2015-07-30T09:03:41.000-05:00,involve generic types,"When an association between documents involves generic types, the type information is not correctly inferred at startup time resulting in missing indexes.
see https://github.com/agustisanchez/SpringDataMongoDBBug for code samples
Given:
attribute super class AbstractProduct with index
When defining a class Catalog with a list of ""wrapped"" books:
The index ""name"" inherited from AbstractProduct is created (book2.content.name) inside ""catalog"" , but the index defined on the Book class itself (isbn) is not created as Spring Data Mongo is only inferring type infromation from the ProductWrapper class definition (ProductWrapper <T extends AbstractProduct>).
If the wrapper class is defined as ProductWrapper<T>, then no indexes are created at all on Catalog.books2.content.","org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexResolver
org.springframework.data.mongodb.core.index.MongoPersistentEntityIndexResolverUnitTests"
FILE,DATAMONGO,DATAMONGO-1289,2015-09-17T11:43:51.000-05:00,"NullPointerException when saving an object with no ""id"" field or @Id annotation","Instead of this, I'm seeing a NullPointerException when attempting to save an object that doesn't have an ""id"" field or an @Id annotated field.
work in 1.6.0 not work in 1.6.0 change in 1.6.2 expect in newer versions","org.springframework.data.mongodb.repository.support.MongoRepositoryFactory
org.springframework.data.mongodb.repository.support.MappingMongoEntityInformation"
FILE,DATAMONGO,DATAMONGO-1360,2016-01-16T07:47:34.000-06:00,not query with JSR310,"I have a MongoDb document I successfully store using Spring Data MongoDb.
It looks like this:
When I create a custom Criteria query that looks like this:
The resulting MongoDb query looks like this:
It consequently fails with this message:
not serialize class java.time.LocalDateTime at org.bson.BasicBSONEncoder.
It does not fail when I use a java.util.Date in my query even though I have stilled persisted my document with a java.time.LocalDateTime object.
The query then looks slightly different like this:
convert LocalDateTime objects to Date objects","org.springframework.data.mongodb.core.Venue
org.springframework.data.mongodb.core.geo.AbstractGeoSpatialTests
org.springframework.data.mongodb.core.MongoTemplate"
FILE,DATAMONGO,DATAMONGO-1438,2016-05-26T14:01:14.000-05:00,I get a warning in my logs since switched to Spring Data MongoDB Hopper-SR1 Release Train in Spring Boot 1.3.5,"When I start my Spring Boot 1.3.5 application with no custom conversions and with Spring Data MongoDB Release Train Hopper-SR1 I get following warning in my logs:
register converter from class java.lang.Number register converter to class write converter support type for registering not convert to Mongo
check annotation setup at converter implementation
I have alle my Domain classes they are saved in MongoDB annotated with @Document (see DATAMONGO-1413)","org.springframework.data.mongodb.core.convert.MongoConvertersUnitTests
org.springframework.data.mongodb.core.convert.MongoConverters"
FILE,DATAMONGO,DATAMONGO-1406,2016-04-04T18:59:49.000-05:00,Query mapper does not use @Field field name when querying nested fields in combination with nested keywords,"we have a document class;
In MongoTemplate.java, the call to :
protect <s, t> List<T>
resolves the fields to the input query to the ones in the @Field annotations, except for these in embedded arrays.
So, in the example above, resolution fields in DBObject remains resolution.
While, the status field resolves to stat.
note queries in inner list
The query submitted to mongo after getMappedObject is called:
Which doesn't get any data, because there is no field called resolution (the field in mongo is res).
Note: The query input to getMappedObject is:
Notice the status and displays fields correctly get converted to the value in the @Field annotation.
This basically means that any queries that operate on fields (with a name different from the peristed name) in the inner list will fail.","org.springframework.data.mongodb.core.convert.QueryMapper
org.springframework.data.mongodb.core.convert.QueryMapperUnitTests"
CLASS,derby-10.7.1.1,DERBY-4684,2010-06-01T10:46:19.000-05:00,correct implicit casting of other types correct implicit casting to BOOLEAN,"Via a UNION query, it is possible to implicitly cast non-BOOLEAN types to BOOLEAN today.
These implicit casts do not obey the SQL Standard rules defined in part 2, section 6.12 (<cast specification>).
The following query shows how you can implicitly cast a non-BOOLEAN data type to BOOLEAN today:
select isindex from sys.sysconglomerates","org.apache.derby.iapi.types.SQLBoolean
org.apache.derby.impl.sql.compile.BooleanTypeCompiler"
CLASS,derby-10.7.1.1,DERBY-4693,2010-06-05T06:36:41.000-05:00,column rename looses 'identity' attribute.,"renaming an identity column causes the column to loose its 'identity' attribute.
you can reproduce the problem as follows:
insert into njr_version values
rename column njr_version
not accept NULL value
To compound this problem there is no way to restore the renamed column's identity attribute as the alter command doesn't let you set an identity attribute.
create new table copy across data",org.apache.derby.impl.sql.execute.RenameConstantAction
CLASS,derby-10.7.1.1,DERBY-4835,2010-10-06T11:05:13.000-05:00,Trigger plan does not recompile with upgrade from 10.5.3.0 to 10.6.1.0 causing  java.lang.NoSuchMethodError,"Trigger plan does not recompile on upgrade from 10.5.3.0 to 10.6.1.0  causing the following exception  the first time the trigger is fired after upgrade.
To reproduce, run the attached script 10_5_3_work.sql with the 10.5.3.0  release and then connect with 10.6.1.0 and insert into the table with the trigger:
have code in handleMinorRevisionChange have DERBY-1107 change in handleMinorRevisionChange relate to DERBY-1107 change relate if bootingDictionary.updateMetadataSPSes(tc)
not be in else clause
recreate trigger work after connecting work around issue drop after connecting drop around issue connect with 10.6.1","org.apache.derby.impl.sql.catalog.DD_Version
org.apache.derbyTesting.functionTests.tests.upgradeTests.BasicSetup"
CLASS,derby-10.7.1.1,DERBY-4889,2010-11-05T20:06:56.000-05:00,Different byte to boolean conversion on embedded and client,"The following code prints ""true"" with the embedded driver and ""false"" with the client driver:
cast (
If setByte() is replaced with setInt(), they both print ""true"".","org.apache.derbyTesting.functionTests.tests.jdbcapi.ParameterMappingTest
org.apache.derby.impl.drda.DRDAConnThread"
CLASS,pig-0.11.1,PIG-2767,2012-06-25T09:11:20.000-05:00,Pig creates wrong schema after dereferencing nested tuple fields,"The following script fails:
DESCRIBE thinks it is {f1: int, f2: int} instead.
When dump is
used, the data is actually in form of the correct schema however, ex.
Because the schema is incorrect,
the reference to ""nested_tuple"" in the ""uses_dereferenced"" statement is
considered to be invalid, and the script fails to run.
The error is:
invalid field projection
not exist in schema","src.org.apache.pig.newplan.logical.expression.DereferenceExpression
test.org.apache.pig.test.TestPigServer"
CLASS,pig-0.11.1,PIG-3114,2013-01-03T19:49:42.000-06:00,Duplicated macro name error when using pigunit,"I'm using PigUnit to test a pig script within which a macro is defined.
Pig runs fine on cluster but getting parsing error with pigunit.
try basic pig script with macro try basic pig script with getting macro similar error get similar error
parse <line 9> null
Pig script which is failing :
remove macro pigunit
define macro parse error result in parsing","src.org.apache.pig.PigServer
test.org.apache.pig.test.pigunit.TestPigTest
test.org.apache.pig.pigunit.PigTest
test.org.apache.pig.pigunit.pig.PigServer"
CLASS,pig-0.11.1,PIG-3267,2013-04-03T16:14:30.000-05:00,fail in limit query,"The following query fail:
generate age into using org.apache.hcatalog.pig.HCatStorer generate age as number
Error happens before launching the second job. Error message:","src.org.apache.pig.backend.hadoop.executionengine.physicalLayer.relationalOperators.POStore
test.org.apache.pig.test.TestMRCompiler"
CLASS,pig-0.11.1,PIG-3292,2013-04-24T03:06:41.000-05:00,Logical plan invalid state: duplicate uid in schema during self-join to get cross product,"work in different way
release note include into CDH dist http://archive.cloudera.com/cdh4/cdh/4/pig-0.10.0-cdh4.2.0.CHANGES.txt
The problem:
We want to do self join to get cross-product
{code}
a = load '/input' as (key, x);
generate flatten(pair)
And an error:
{code}
ERROR org.apache.pig.tools.grunt.Grunt - ERROR 2270: Logical plan invalid state: duplicate uid in schema : 1-7::x#16:bytearray,y::x#16:bytearray
{code}
generate flatten(pair)","test.org.apache.pig.test.TestEvalPipelineLocal
src.org.apache.pig.newplan.logical.relational.LOCross"
CLASS,pig-0.11.1,PIG-3310,2013-05-03T02:59:57.000-05:00,"ImplicitSplitInserter does not generate new uids for nested schema fields, leading to miscomputations","Consider the following example
inp AS
j by shopId
provide minimal reproduction case
On input data:
{code}
1       1001    101
1       1002    103
1       1003    102
1       1004    102
2       1005    101
2       1003    101
2       1002    123
3       1042    101
3       1005    101
3       1002    133
{code}
This will give a wrongful output like .
.
{code}
(1 1001,1001)
(1 1002,1002)
(1 1002,1002)
(1 1002,1002)
{code}
move LOFilter operation before join be in initial case fail at place fail because_of PushUpFilter optimization work on tuple
create LOSplitOutputs reset schema regenerate uids for fields
get new uid
have same uid have join looks
get separate uids
recurse on nested schema fields
have light understanding
reproduce issue
run unit tests with fix
like wrong way fix issue",src.org.apache.pig.newplan.logical.relational.LOSplitOutput
CLASS,pig-0.11.1,PIG-3329,2013-05-16T22:44:41.000-05:00,work with SPLIT,"input.txt:
1 2 3
4 5 6
7 8 9
script:
a = load 'input.txt' using PigStorage(' ') as (a:int, b:int, c:int);
SPLIT a into b if a > 0, c if a > 5;
d = RANK b;
dump d;
job will fail with error message:
read counter pig.counters.counter_4929375455335572575_-1","src.org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.MapReduceOper
src.org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.MRCompiler"
CLASS,pig-0.11.1,PIG-3379,2013-07-16T13:37:26.000-05:00,Alias reuse in nested foreach causes PIG script to fail,"The following script fails:
not exist in schema
use distinct alias name for 2nd distinctdevices
remove last filter statement fix as observation","src.org.apache.pig.parser.LogicalPlanBuilder
src.org.apache.pig.PigServer
test.org.apache.pig.test.TestEvalPipeline2
src.org.apache.pig.newplan.logical.relational.LOInnerLoad
src.org.apache.pig.newplan.logical.expression.ProjectExpression"
CLASS,mahout-0.8,MAHOUT-1285,2013-07-15T16:31:28.000-05:00,Arff loader can misparse string data as double,"Have successfully loaded numerous ARFF files with Mahout (originally generated via WEKA).
generate data
For a specific random seed, the following exception is thrown:","integration.src.main.java.org.apache.mahout.utils.vectors.arff.ARFFVectorIterable
integration.src.main.java.org.apache.mahout.utils.vectors.arff.MapBackedARFFModel"
CLASS,mahout-0.8,MAHOUT-1314,2013-08-18T09:07:48.000-05:00,StreamingKMeansReducer throws NullPointerException when REDUCE_STREAMING_KMEANS is set to true,"when REDUCE_STREAMING_KMEANS option is set to true (-rskm) the reducer fails with NullPointerException.
reduce method on line
full stack trace:
it happens every time the REDUCE_STREAMING_KMEANS is set to true.",core.src.main.java.org.apache.mahout.clustering.streaming.mapreduce.StreamingKMeansReducer
CLASS,mahout-0.8,MAHOUT-1358,2013-11-18T01:58:22.000-06:00,StreamingKMeansThread throws IllegalArgumentException when REDUCE_STREAMING_KMEANS is set to true,"Running StreamingKMeans Clustering with REDUCE_STREAMING_KMEANS = true and when no estimatedDistanceCutoff is specified, throws the following error
have nonzero number of training test vectors
ask for %
The issue is caused by the following code in StreamingKMeansThread.call()
use same iterator fail on second use",core.src.main.java.org.apache.mahout.clustering.streaming.mapreduce.StreamingKMeansThread
CLASS,zookeeper-3.4.5,ZOOKEEPER-1179,2011-09-13T11:20:56.000-05:00,NettyServerCnxn does not properly close socket on 4 letter word requests,"When calling a 4-letter-word to a server configured to use NettyServerCnxnFactory, the factory will not properly cancel all the keys and close the socket after sending the response for the 4lw.
The close request will throw this exception, and the thread will not shut down:
not complete java.io.IOException not complete WARN","src.java.main.org.apache.zookeeper.server.NettyServerCnxn
src.java.main.org.apache.zookeeper.server.NettyServerCnxnFactory"
CLASS,zookeeper-3.4.5,ZOOKEEPER-1781,2013-10-03T20:19:27.000-05:00,ZooKeeper Server fails if snapCount is set to 1,"If snapCount is set to 1, ZooKeeper Server can start but it fails with the below error:
take snapshot at same time
mention restriction in documentation add validation in source code",src.java.main.org.apache.zookeeper.server.ZooKeeperServer
CLASS,argouml-0.22,3100,2005-02-22T00:28:44.000-06:00,new package or class shown with funny name in explorer,"After adding a new package, the package is shown in the explorer with a funny 
name: ""(anon Package)"" or ""(anon Class)"".
stand for anonymous package stand for anonymous class come after many hours come after thinking","org.argouml.uml.ui.UMLListCellRenderer2
org.argouml.ui.DisplayTextTree"
CLASS,argouml-0.22,3774,2005-12-03T06:33:15.000-06:00,not set namespace of Model add to diagram,"To repeat:
Start argouml.
Drag the created defaultmodel into the class diagram.
Here is the backtrace:
echo --- executing ArgoUML ---
find [ argouml ] log4j:WARN for logger
argouml ] log4j:WARN initialize log4j system
argouml ] javax.jmi.reflect.CompositionCycleException",org.argouml.uml.diagram.static_structure.ClassDiagramGraphModel
CLASS,argouml-0.22,3923,2006-02-07T13:17:48.000-06:00,import Poseidon activity diagrams from XMI,"use activity diagram for AndroMDA
1) Import an XMI from Poseidon, which works well with AndroMDA (the
PiggyBank example).
2) If I add my activity diagram under the use case diagram I always get a new activity graph, so I have 2 activity graphs alltogether.
I cannot add an activity diagram under the imported activity graph.
Please see the screenshot I attached.
Screenshot:
3) This code, which works with Poseidon, won't work with ArgoUML:
actionState is always ""null"".
4) Importing the activity diagram from Poseidon works and the result can be processed by AndroMDA but if you are making the activity diagram from the beginning with ArgoUML, it won't work because of the error above",org.argouml.persistence.XMIParser
CLASS,argouml-0.22,3945,2006-02-13T04:04:40.000-06:00,Exception assigning an operation to a callevent,"The following Exception happens when assigning an operatioon to a callevent in 
the following way:
1 Start ArgoUML.
2 Draw a class.
3 Create an operation for the class.
4 Select the class, and create a statechart diagram.
5 Create 2 states and a transition.
6 Rightclick the trigger fiels of the transition and create a new callevent.
7 Create a parameter for this callevent.
8 Set the type of the parameter to int and the initial value to 1, and give
it a name.
9 Navigate to the callevent, and select the ""newOperation"" from the dropdown.
Bang:
occur ************ feb CET [ argouml ] java.lang.IllegalArgumentException handle callevent$Impl ID
fe OPCKG",org.argouml.uml.ui.behavior.state_machines.UMLCallEventOperationComboBox2
CLASS,argouml-0.22,3974,2006-02-19T03:58:40.000-06:00,Methods and properties not updated in class diagrams,"If I change the name of a method or property in a class diagrams, it won't be
updated until I reload the diagram, or add a parameter.
The same thing happens
if I change the visibility or modifier (static for example) of a method or property.
use last version",org.argouml.uml.diagram.static_structure.ui.FigClass
CLASS,argouml-0.22,4037,2006-03-06T11:33:42.000-06:00,Excessive redraws in some diagrams in CVS version (regression from 0.20),"I see ArgoUML taking upwards of 40% of my processor time on some diagrams.
For instance, I have one project in which the two most complicated class
diagrams burn 50% or so of my processor time.
Running the Quartz Debug application, I see that on those two diagrams the
diagram is being entirely repainted constantly.
This is not the case on any of
the other diagrams.
have big screen take bit of time
So I went back to the 0.20 release version, and I don't see this behavior (in
fact, the CPU load is better even on the other drawings).
build CVS version use build.sh package
file to proper place
examine things cause redrawing for examining cause complexity for examining
see in attached ca3Design.zargo file
I have made a simple diagram with one association that triggers this behavior.
Look at the ""Redraw Problem Demonstrated"" class diagram.
If you open the context menu on the Association and choose
""Navigability/bidirectional"" the constant repainting stops.",org.argouml.uml.diagram.ui.FigAssociation
CLASS,argouml-0.22,4085,2006-03-13T10:43:14.000-06:00,Exception adding deferrable event in Action State,"An IllegalArgumentException is thrown when you select from the Deferrable Events
pane of the Action State property panel when you choose Select -> Add... from
the context (popup) menu.
object [ actionstate$Impl ID
eb MID",org.argouml.uml.ui.behavior.state_machines.PopupMenuNewEvent
CLASS,argouml-0.22,4103,2006-03-19T08:42:23.000-06:00,Exception when deleting transition,"The following causes an exception:
1 Create a statechart diagram.with 2 states and a transition.
2 Make the transition visible in the explorer.
3 Delete one of the states.
mar CET 2006",org.argouml.ui.DisplayTextTree
CLASS,argouml-0.22,4245,2006-06-04T13:09:07.000-05:00,Repaint exception when creating sequence diagram,"I get the following exception the first time a sequence diagram is created.
catch in debugger look at context
report in comments
thread AWT-eventqueue-0 java.lang.NullPointerException",org.argouml.uml.diagram.ui.TabDiagram
CLASS,argouml-0.22,4330,2006-07-11T06:48:26.000-05:00,"Exception in ""Send email to expert""","Steps to reproduce:
- select an active critic from the ToDoPane
- press the ""send email to expert"" button
The following exception is thrown (stack from the console, no pop-ups appear):
thread AWT-eventqueue-0 java.lang.NullPointerException",org.argouml.cognitive.ui.TabToDo
CLASS,argouml-0.22,4349,2006-07-15T12:49:43.000-05:00,Activity Diagram not refreshed after deletion of a Action State,"repeat bug
1 Delete all the transiton associted with the action state you want to delete.
2 Delete the action state.
Nothing happens the action state does not get removed form the Activity Diagram
and as soon as you save the model.
The action state does get remove and the
following error is generated.
save model.These
delete Action state","org.argouml.uml.diagram.activity.ui.FigActionState
org.argouml.uml.diagram.collaboration.ui.FigClassifierRole
org.argouml.uml.diagram.activity.ui.FigObjectFlowState
org.argouml.uml.diagram.sequence.ui.FigClassifierRole
org.argouml.uml.diagram.activity.ui.FigCallState
org.argouml.uml.diagram.state.ui.FigSubmachineState"
CLASS,argouml-0.22,4362,2006-07-18T12:07:56.000-05:00,Exception thrown drawing generalization,"Place 2 collaborations and draw a generalization between them, an exception is
thrown.
commit similar fix",org.argouml.uml.diagram.collaboration.ui.CollabDiagramRenderer
CLASS,argouml-0.22,4364,2006-07-18T15:30:05.000-05:00,Dragging association onto diagram creates n-ary diamond,"There's also an exception thrown.
1 Start ArgoUML
2 Place 2 classes and association them
3 Drag the association from explorer onto diagram
add edge of type org.omg.uml.foundation.core.AssociationEnd$Impl","org.argouml.uml.diagram.collaboration.CollabDiagramGraphModel
org.argouml.uml.diagram.use_case.UseCaseDiagramGraphModel
org.argouml.uml.diagram.static_structure.ClassDiagramGraphModel
org.argouml.uml.diagram.deployment.DeploymentDiagramGraphModel"
CLASS,argouml-0.22,4367,2006-07-19T19:00:56.000-05:00,deleting call self action throws InvalidElementException and makes Argo unstable,"steps:
1 - create a *new file*
2 - create a sequence diagram
3 - create 2 classifier roles
4 - create new call action from 1st to 2nd
5 - create new call action from 2nd to it self
6 - assign a name to calls
7 - select call action in step 5
8 - Right click and select delete from model
At this time Argo is complete unstable
9 - Save & get a report (0.22b2 only)
find in 0.22B1 find in b2","org.argouml.uml.diagram.sequence.ui.FigClassifierRole
org.argouml.uml.diagram.sequence.ui.SequenceDiagramLayer"
CLASS,lucene-4.0,LUCENE-4461,2012-10-05T10:21:38.000-05:00,Multiple FacetRequest with the same path creates inconsistent results,"Multiple FacetRequest are getting merged into one creating wrong results in this case:
define hashcode equal in certain way talk about different requests
test case",org.apache.lucene.facet.search.StandardFacetsAccumulator
METHOD,eclipse-2.0,19686,2002-06-07T16:42:00.000-05:00,mouse over end of file work after mousing,"build from Stable
editor in java
Move the mouse to the whitespace at the end of 
the file.
like lower-right-most region of java file
Stay there for 
about 3 secs or so, then mouse over some text in the java file where you would 
expect hover help.
The hover help no longer works for the file.
help button on/off/on bring hoverhelp switch to other editors
close file reopen file get hover help",org.eclipse.jface.text.TextViewerHoverManager:computeInformation()
METHOD,eclipse-2.0,19976,2002-06-11T22:25:00.000-05:00,not add bad site modify site properties,"If in Site Explorer you add a new site with a bogus url, the site shouldn't get 
added.
At present hitting <Finish> will inform you of the error but the site 
will still get added.
argue as feature not connect to internet
do with CVS do with offer",org.eclipse.team.internal.webdav.ui.WebDavConfigurationWizard:performFinish()
METHOD,eclipse-2.0,23140,2002-09-04T01:34:00.000-05:00,man in middle attacks poisoning,"use shared codebase test on Linux gtk2 test on mac OS x
base ssh1 client with cvs base ssh1 client for use
select extssh as method select extssh to cvs repository
support password authentication use standard ext method with suitable ssh-agent use standard ext method for public key authentication
However when using the Eclipse extssh component no history of host keys
is kept.
enable attacker poison name resolution of eclipse client poison attacker of eclipse client steal passwords through modified ssh daemon
1 Set up access to an external cvs repository with extssh and a
password.
(preferably with a username that doesn't exist locally eg -
notme).
Close eclipse.
2 Edit /etc/hosts to point to another ssh server (eg localhost)
3 Open eclipse and choose to browse the repository from the cvs
perspective.
4 In your auth logs you should get messages like:
log in case
gain access to valuable accounts
use ssh2 client with public keys","org.eclipse.team.internal.ccvs.ssh.Client:connect(IProgressMonitor)
org.eclipse.team.internal.ccvs.ssh.Client:send_SSH_CMSG_SESSION_KEY(byte[], byte[], byte[], byte[], byte[], byte[])
org.eclipse.team.internal.ccvs.ssh.Client:receive_SSH_SMSG_PUBLIC_KEY(ServerPacket)"
METHOD,eclipse-2.0,31779,2003-02-13T09:55:00.000-06:00,ensure [ resources,"use natives
When the UnifiedTree finds a new file from the file system, it assumes that if the file is not an existing file, then it is a folder.
This is not always true, because for different reasons a file returned by java.io.File.list/listFiles may not actually exist (our CoreFileSystemLibrary#getStat() returns 0).
This problem appears to the user when executing refresh operations.
At the first moment, the file is found in the file system and assumed to be a folder, and a corresponding resource is created in the workspace.
At the second refresh, the folder corresponding to that resource is not found in the file system, and then it is removed from the workspace.
And so on.
reveal bugs","org.eclipse.core.internal.localstore.UnifiedTree:addChildrenFromFileSystem(UnifiedTreeNode, String, Object[], int)
org.eclipse.core.internal.localstore.UnifiedTree:createChildNodeFromFileSystem(UnifiedTreeNode, String, String)"
CLASS,jabref-2.6,1436014,2006-02-21T16:27:48.000-06:00,No comma added to separate keywords,"When adding a keyword via the content selector, no comma is added before the keyword.
I have set ""When adding/removing keywords separate them by:"" to "", "" in the Preferences > Groups panel.
But keywords still get separated by a space only.
apply to JabRef 2.0.1","net.sf.jabref.groups.GroupDialog
net.sf.jabref.FieldContentSelector
net.sf.jabref.EntryEditor"
CLASS,jabref-2.6,1631548,2007-01-09T14:20:57.000-06:00,"""Open last edited DB at startup"" depends on the working dir","another little bug/feature: The JabRef option ""Open last edited database at startup"" depends on the working directory at which JabRef is started.
Example:
results in an empty JabRef not opening $\{HOME\}/at-work/Bibliography/my\_documents.bib.
store absolute path for open last edited
store dependent configuration file \ ( ~ / in machine store dependent configuration file \ ( ~ / since setting
store relative path store absolute path store home directory try relative path try absolute path try home directory open JabRef
regard bernd
find bug in JabRef",net.sf.jabref.JabRefFrame
CLASS,jabref-2.6,1709449,2007-04-29T00:03:55.000-05:00,Clicking a DOI from context menu fails,"have document with DOI have document with hyperlink
show \ ( nice \ in document list show \ ( nice \ in offers
Now: Clicking the DOI icon fails, since the default DOI server is not prepended, i.e. Firefox says
retrieve URL
save DOI for document save hyperlink for document","tests.net.sf.jabref.UtilTest
net.sf.jabref.Util"
CLASS,jabref-2.6,2207462,2008-10-29T14:31:16.000-05:00,Web search results hidden behind JabRef window,"When searching for a string with the ""Web Search > Search ACM Digital Library"" function, the following happens \(e.g. searching ""sql injection"" in The Guide\):
\* A new window opens with progress bar where the results will later be shown \(yet empty\).
\* The new window disappears \(behind the main JabRef window\) and a message box opens: ""444 entries found.
To reduce server load, only 50 will be downloaded.""
\* Hitting ok: Message box closes, but the result window stays in the background \(invisible\).
move JabRef window find results","net.sf.jabref.imports.ACMPortalFetcher
net.sf.jabref.imports.ImportInspectionCommandLine
net.sf.jabref.gui.ImportInspectionDialog"
CLASS,jabref-2.6,2938562,2010-01-24T12:59:46.000-06:00,Move of files turns relative filenames to absolute ones,"The ""general""-tab provides the management of files attached to a bib entry.
The context-menu of files in this list provides to move/rename functions.
Using this turns relative filenames \(relative to the doc-folder of the database settings\) to absolute filenames.
observe in JabRef 2.6b2",net.sf.jabref.external.MoveFileAction
METHOD,mahout-0.8,MAHOUT-1301,2013-08-01T09:31:21.000-05:00,toString() method of SequentialAccessSparseVector has excess comma at the end,"change in MAHOUT-1259 patch
Unfortunately, that patch introduced new bug: output of the toString() method had been changed - extra comma added at the end of the string
Example: 
Consider following sparse vector
{code:java}
Vector v = new SequentialAccessSparseVector(capacity);
v.set(1, 0.1);
v.set(3, 0.3);
{code}
In 0.7 v.toString() returns following string:
{code:java}
{1:0.1,3:0.3}
{code}
but in 0.8 it returns
{code:java}
{1:0.1,3:0.3,}
{code}
As you can see, there is extra comma at the end of the string.","org.apache.mahout.math.SequentialAccessSparseVector:toString()
org.apache.mahout.math.RandomAccessSparseVector:toString()"
METHOD,mahout-0.8,MAHOUT-1358,2013-11-18T01:58:22.000-06:00,StreamingKMeansThread throws IllegalArgumentException when REDUCE_STREAMING_KMEANS is set to true,"Running StreamingKMeans Clustering with REDUCE_STREAMING_KMEANS = true and when no estimatedDistanceCutoff is specified, throws the following error
have nonzero number of training test vectors ask for %
cause issue
use same iterator fail on second use","org.apache.mahout.clustering.streaming.mapreduce.StreamingKMeansThread:StreamingKMeansThread(Path, Configuration)
org.apache.mahout.clustering.streaming.mapreduce.StreamingKMeansThread:StreamingKMeansThread(Iterable<Centroid>, Configuration)"
CLASS,openjpa-2.0.1,OPENJPA-1903,2010-12-06T13:05:34.000-06:00,Some queries only work the first time they are executed,"I have a problem in my application where a query that sometimes returns data and sometimes not.
reduce to code project into Eclipse project at http://ubuntuone.com/p/S9n/
happen with OpenJPA 2.0.1 happen with daily snapshot happen from out-of-process Derby database
Basically I have two Entities which both use multiple Ids to produce the Primary Key, ""Preis"" contains a foreign key on ""Website"":
datum }
I use the following to set up a website and a Preis:
Afterwards, if I run the query as follows:
this query works all the time, note that it uses website.name for matching, not the full Website-object.
However if I put the query as
it only works ONCE and then does not return any results any more!!
see testcase DataAccessVerifyTest for details",org.apache.openjpa.jdbc.kernel.PreparedQueryImpl
CLASS,openjpa-2.0.1,OPENJPA-1912,2011-01-03T13:48:09.000-06:00,enhancer generates invalid code if fetch-groups is activated,"If openjpa.DetachState =fetch-groups is used, the enhancer will add a 'implements Externalizable' + writeExternal + readExternal.
The problem is, that writeExternal and readExternal will also try to externalize the private members of any given superclass.
Thus we get a runtime Exception that we are not allowed to access those fields.
Example:
will result in the following code (decompiled with jad):
throw IOException",org.apache.openjpa.enhance.PCEnhancer
CLASS,openjpa-2.0.1,OPENJPA-1918,2011-01-06T08:11:24.000-06:00,MetaDataRepository.preload() ignores class loader returned by PersistenceUnitInfo.getClassLoader(),"use openjpa inside osgi container
We pass the appliation class loeader as part of our PersistenceUnitInfo implementation by returning it from PersistenceUnitInfo.getClassLoader().
However, the code in MetaDataRepository.preload() only uses the context class loader and not the class loader from PersistenceUnitInfo, which leades to ClassNotFoundExpcetions like mentioned at the end of this report.
append return value of PersistenceUnitInfo.getClassLoader() append return value to list establihe fix participate in MultiClassLoader set up in MetaDataRepository.java:310ff
set classloader as context loader set classloader in meanwhile set classloader by PersistenceProvider.createContainerEntityManagerFactory() set classloader during creation
instantiate bean entityManagerFactory of class null
see nested stacktrace for details
cause by org.clazzes.fancymail.server.entities.EMail","org.apache.openjpa.meta.FieldMetaData
org.apache.openjpa.meta.MetaDataRepository
org.apache.openjpa.persistence.detach.NoVersionEntity"
CLASS,openjpa-2.0.1,OPENJPA-1928,2011-01-20T17:43:52.000-06:00,Resolving factory method does not allow method overriding,"If a get method is annotated with @Factory then the method cannot be overridden with a method which take different parameters.
The system randomly selects one of the several methods with the same name which may or may not take the type which will be provided.
For example:
        @Persistent(optional = false)
	@Column(name = ""STATUS"")
	@Externalizer(""getName"")
	@Factory(""valueOf"")
	public OrderStatus getStatus() {
		return this.status;
	}
Actual results:
valueOf(String) may or may not be selected.
fix defect by applying apply method invocation conversion rules from Java",org.apache.openjpa.meta.FieldMetaData
CLASS,openjpa-2.0.1,OPENJPA-2007,2011-05-25T14:51:51.000-05:00,"Setting query hint ""openjpa.FetchPlan.Isolation"" results in misleading warning","Setting query hint ""openjpa.FetchPlan.Isolation"" on a query results in this misleading warning being printed in the log:
mean openjpa.FetchPlan.LRSSize
The supported hint map in JDBCPersistenceProductDerivation adds key openjpa.FetchPlan.TransactionIsolation, which is incorrect.
Note: Even though a warning is issued, provided the database supports query-level isolation level switching, the hint handler does apply the isolation level to the query.","org.apache.openjpa.persistence.jdbc.JDBCPersistenceProductDerivation
org.apache.openjpa.conf.TestQueryHints"
METHOD,lang,LANG-477,2009-01-09T10:05:53.000-06:00,ExtendedMessageFormat: OutOfMemory with custom format registry and a pattern containing single quotes,"When using ExtendedMessageFormat with a custom format registry and a pattern conatining single quotes, an OutOfMemoryError will occur.
Example that will cause error:
start at line start on release","org.apache.commons.lang.text.ExtendedMessageFormat:appendQuotedString(String, ParsePosition, StringBuffer, boolean)"
METHOD,lang,LANG-480,2009-01-20T17:36:44.000-06:00,StringEscapeUtils.escapeHtml incorrectly converts unicode characters above U+00FFFF into 2 characters,"Characters that are represented as a 2 characters internaly by java are incorrectly converted by the function.
The following test displays the problem quite nicely:
import org.apache.commons.lang.
throw Exception { //
want patch","org.apache.commons.lang.Entities:escape(Writer, String)"
METHOD,lang,LANG-538,2009-10-16T16:47:39.000-05:00,DateFormatUtils.format does not correctly change Calendar TimeZone in certain situations,"If a Calendar object is constructed in certain ways a call to Calendar.setTimeZone does not correctly change the Calendars fields.
For example, the following unit test fails:
construct with cal = construct with GMT-8 default locale time zone GregorianCalendar cal work in time zone
However, this unit test passes:","org.apache.commons.lang3.time.FastDateFormat:format(Calendar, StringBuffer)"
METHOD,lang,LANG-788,2012-02-11T12:36:48.000-06:00,SerializationUtils throws ClassNotFoundException when cloning primitive classes,"If a serializable object contains a reference to a primitive class, e.g. int.class or int[].class, the SerializationUtils throw a ClassNotFoundException when trying to clone that object.
noformat } import org.apache.commons.lang3.SerializationUtils;
import org.junit.Test;
fix java bug http://bugs.sun.com/view_bug.do?bug_id=4171142 since java version fix ObjectInputStream since java version
The SerializationUtils problem arises because the SerializationUtils internally use the ClassLoaderAwareObjectInputStream that overrides the ObjectInputStream's resoleClass method without delegating to the super method in case of a ClassNotFoundException.
For example:
throw IOException {
try { return Class.forName(name, false, classLoader)
try { return Class.forName ( name
catch { return super.resolveClass(desc)
fix java bug in ObjectInputStream fix code in ObjectInputStream
protect class <?> resolveClass(ObjectStreamClass desc)
try { return Class.forName","org.apache.commons.lang3.SerializationUtils:ClassLoaderAwareObjectInputStream(InputStream, ClassLoader)
org.apache.commons.lang3.SerializationUtils:resolveClass(ObjectStreamClass)"
METHOD,lang,LANG-879,2013-03-18T21:46:29.000-05:00,"LocaleUtils test fails with new Locale ""ja_JP_JP_#u-ca-japanese"" of JDK7","The Test below fails with the following error on JDK7, but succeeds on JDK6:
remove stack frames
import static org.hamcrest.Matchers.equalTo;
import java.util.Locale;
import org.testng.annotations.Test;
import com.scispike.foundation.i18n.StringToLocaleConverter;
import java.util.Locale;
import org.springframework.core.convert.converter.Converter;",org.apache.commons.lang3.LocaleUtils:toLocale(String)
FILE,SWARM,SWARM-485,2016-05-26T18:01:46.000-05:00,Using a JAX-RS provider that extends JacksonJaxbJsonProvider throws a  java.lang.NoClassDefFoundError Link Error,"A java.lang.NoClassDefFoundError Link Error is thrown when deploying a JAX-RS application with a custom provider that overrides com.fasterxml.jackson.jaxrs.json.JacksonJaxbJsonProvider
introduce issue in 1.0.0
not occur in 1.0.0",org.wildfly.swarm.jaxrs.JAXRSArquillianTest
FILE,SWARM,SWARM-486,2016-05-28T18:25:37.000-05:00,Can't load project-stages.yml on classpath with Arq,"problem project-stages.
yml on classpath(src/main/resources) is not loaded with Arquillian tests.
I attached the error log and the reproducer in 'Steps to Reproduce' section.
Though -swarm try to load it, apparently can't see it when Arq tests.
load yml",org.wildfly.swarm.container.ProjectStagesTest
FILE,SWARM,SWARM-757,2016-10-06T11:58:09.000-05:00,javax.persistence.PersistenceException: [PersistenceUnit: primary] Unable to build Hibernate SessionFactory when using Infinispan as 2nd level cache for hibernate,"I am trying to enable Infinispan as a 2nd level cache for hibernate.
As soon as I enable it, I get the following startup exception:
start service jboss.persistenceunit.
build Hibernate SessionFactory
build Hibernate SessionFactory cause by javax.persistence.PersistenceException
request service [ org.hibernate.engine.spi.CacheImplementor] cause by org.hibernate.service.spi.ServiceException unable
cause by org.hibernate.cache.CacheException
build Hibernate SessionFactory
build Hibernate SessionFactory cause by javax.persistence.PersistenceException
request service [ org.hibernate.engine.spi.CacheImplementor] cause by org.hibernate.service.spi.ServiceException unable
cause org.hibernate.cache.CacheException
roll back ERROR with following failure message
build Hibernate SessionFactory
build Hibernate SessionFactory cause by javax.persistence.PersistenceException
request service [ org.hibernate.engine.spi.CacheImplementor] cause by org.hibernate.service.spi.ServiceException unable
cause org.hibernate.cache.CacheException",org.wildfly.swarm.infinispan.runtime.InfinispanCustomizer
FILE,SWARM,SWARM-793,2016-10-25T19:04:09.000-05:00,Cannot add custom modules to WildFly Swarm using <resource-root> instead of <artifact> in the module.xml,"use <resource-root> instead_of <artifact> add as modules
Custom module ""oracle.jdbc"" (module.xml and JAR) copied into Uber-JAR's ""modules\oracle\jdbc\main"" directory (there are also some default modules there - jboss, javax, etc) when module is configured like this:
run app get error
find driver class
configure in module.xml
JAR is copied to ""m2repo\com\hynnet\oracle-driver-ojdbc6\12.1.0.1"" directory of Uber-JAR and driver was successfully loaded.
Looks like WFS adding to classpath only what is in ""m2repo"", but not the ""oracle"" module in ""modules"" directory.
miss command-line parameter like jboss_modulepath define location of custom modules","org.wildfly.swarm.bootstrap.modules.NestedJarResourceLoader
org.wildfly.swarm.bootstrap.modules.BootstrapClasspathModuleFinder"
METHOD,derby-10.9.1.0,DERBY-5911,2012-08-28T06:18:36.000-05:00,WHERE condition getting pushed into sub-query with FETCH,"Derby pushes query conditions down into subqueries with FETCH limits, thus creating wrong results.
Take the following snippet:
fetch next rows
The first query correctly returns the rows (Colombian,5), (Colombian_Decaf,20).
The second query (which filters the result of the first one) returns (Colombian,5), (French_Roast,5).
evaluate filter condition before fetch limit","org.apache.derby.impl.sql.compile.SelectNode:checkNoWindowFunctions(QueryTreeNode, String)"
METHOD,derby-10.9.1.0,DERBY-5951,2012-10-16T10:33:53.000-05:00,Missing method exception raised when using Clobs with territory based collation,"When using territory-based collation with Clobs, Derby raises an error trying to invoke a missing method.
The following script shows this problem:
connect jdbc:derby:memory:db
create function makeClob( contents varchar( 32672 ) ) returns
create table clobTable( a clob )
-- fails with a java.lang.NoSuchMethodError exception
insert into clobTable( a ) values
connect jdbc:derby:memory:db1
create function makeClob( contents varchar( 32672 ) ) returns
create table clobTable( a clob )
insert into clobTable( a ) values
Here is the error:
evaluate expression throw org.apache.derby.iapi.types.DataValueFactory.getClobDataValue while evaluating
fail with java.lang.NoSuchMethodError exception
insert into clobTable( a ) values
evaluate expression throw org.apache.derby.iapi.types.DataValueFactory.getClobDataValue while evaluating
cause java.lang.NoSuchMethodError",org.apache.derby.iapi.types.CollatorSQLClob:getNewNull()
METHOD,derby-10.9.1.0,DERBY-6009,2012-12-07T04:15:23.000-06:00,need stricter checking of ORDER BY clause,"support column numbers in ORDER BY clauses see by error message
support unqualified column references support column position numbers
However, the checks let some unsupported expressions through and produce strange results.
For example:
And the next example should probably have raised a syntax error too, instead of a NullPointerException:",org.apache.derby.impl.sql.compile.OrderByList:getResultSetNumber()
METHOD,derby-10.9.1.0,DERBY-6027,2012-12-29T20:41:18.000-06:00,ORDER BY a cast expression gives NPE,"The following gives a NPE:
value by int
or
pull up by expression",org.apache.derby.impl.sql.compile.OrderByColumn:isReferedColByNum(ValueNode)
FILE,eclipse-3.1,100137,2005-06-15T04:29:00.000-05:00,not work in details pane,"1. create a fresh Java project
2. delete the JRE System Library
3. add a new User Library which is marked as system library and has the rt.jar as single library (I used JDK 1.5.0_03)
4. add the following class:
5. add breakpoint on line 8
6. debug
show correct values
- source is not found
not work in detail pane",org.eclipse.jdt.launching.StandardClasspathProvider
FILE,eclipse-3.1,100336,2005-06-15T20:11:00.000-05:00,[EditorMgmt] Default editor ignored when opening file,"find in 3.1rc2
When opening a .java file for the first time, the default editor is ignored and the file is opened with the Colorer Editor (plugin).
mark as default
not find same outcome with other file types
open file continue file use Java editor select Java editor from Open","org.eclipse.ui.internal.dialogs.FileEditorsPreferencePage
org.eclipse.ui.internal.IWorkbenchConstants
org.eclipse.ui.internal.registry.FileEditorMapping"
FILE,eclipse-3.1,100799,2005-06-20T08:40:00.000-05:00,create error page link to error log view,"I compiled a plug-in with a view against a 5.0 JRE.
When I ran this plugin in a run-time eclipse with a 1.4 VM, startup passed without any error message.
When I opened the view, I got a view with an embedded error dialog showing a
ClassNotFoundException.
But the real reason was in the previous log entry (an UnsupportedClassVersionError).
create view ID org.eclipse.jdt.jeview.views.JavaElementView load class org.eclipse.jdt.jeview.views.JavaElementView.java.lang.ClassNotFoundException
activate bundle org.eclipse.jdt.jeview","org.eclipse.ui.internal.WorkbenchMessages
org.eclipse.ui.internal.part.StatusPart"
FILE,eclipse-3.1,100808,2005-06-20T09:30:00.000-05:00,replace range for package proposals,"see bug for details
- have an empty .java file with the contents below (no line termination)
- invoke code assist at the end of the first line
The reported completion proposals specify a replace range of [8, 13), while the document length is only 12.
note [
Since jdt-ui does not check the replace ranges, we run into the IAE of
StyledText eventually (see bug 100795 for the trace).",org.eclipse.jdt.internal.codeassist.complete.CompletionScanner
FILE,eclipse-3.1,101090,2005-06-21T13:26:00.000-05:00,No console output when running ant in separate jvm,"When I RC->Run as->Ant Build, no stdout or stderr gets written to console.
use same jvm
install with third party plugins install with ~ /
eclipse dir on linux fc4 use j2sdk1.4.2_07","org.eclipse.ant.internal.ui.IAntUIPreferenceConstants
org.eclipse.ant.internal.ui.AntUIPreferenceInitializer
org.eclipse.ant.internal.ui.preferences.AntPreferencePage
org.eclipse.ant.internal.ui.preferences.AntPreferencesMessages
org.eclipse.ant.internal.ui.launchConfigurations.RemoteAntBuildListener"
FILE,eclipse-3.1,101328,2005-06-22T15:13:00.000-05:00,get error commit removal,"Using N0622, select the removal that you want to commit and you will get an error in the log about the fact that the resource does't exist.
commit removed element not exist of course
select parent commit change",org.eclipse.team.internal.ccvs.core.CVSWorkspaceSubscriber
FILE,eclipse-3.1,102473,2005-07-01T09:10:00.000-05:00,code assist: parameter names not harvested from debug info in class files,"3.1 get this zip http://prdownloads.sourceforge.net/jggapi/jggapi-1.0.1-bin.zip?download
extract the jar file from it and put on build path
open the class file editor for LoginContext.
Look at constructor.
You see that both parameter names are known: uin, password
Now, try completing:
new LoginContext(<CODEASSIST>
you get 2 not informative parameter hints: arg0, arg1
assist settings problem
work out_of box","org.eclipse.jdt.internal.core.BinaryMethod
org.eclipse.jdt.internal.compiler.classfmt.MethodInfo"
FILE,eclipse-3.1,103379,2005-07-11T15:37:00.000-05:00,[MPE] [EditorMgmt] An editor instance is being leaked each time an editor is open and closed,"Every we open and close an editor.
That editor instance is being leaked.
have testcase demostrate testcase
It creates a new simple project and a new file.
It opens up the new file in the editor that comes with the testcase, then close the editor.
Repeat 500 times.
What's interesting is that the editor, upon open, will allocate a 200000 size
String array as a private field.
If you run this testcase with -Xmx256M, you will run out of memory.
However, if you explicitly set the String array to null in the dispose() method of the editor, then the same testcase will not run out of memory.
This leads us to believe that the editor instance is being leaked.",org.eclipse.ui.operations.OperationHistoryActionHandler
FILE,eclipse-3.1,103918,2005-07-14T17:25:00.000-05:00,100% CPU load while creating dynamic proxy in rich client app,"integrate ecplipse-rcp application with springframework
I've
noticed that when spring tries to instantiate any dynamic proxy RCP falls into
infinit loop, CPU gets 100% load and the application needs to be killed.
my start method contains the following code:
throw Exception { super.start(context)
xml ) )
When bf.getBean(""hang"") is executed the application hangs.
execute outside eclipse-rcp
create proxy class for given interface
attach sample project",org.eclipse.core.runtime.internal.adaptor.ContextFinder
FILE,eclipse-3.1,104966,2005-07-24T23:17:00.000-05:00,Link displays as enabled when setText called after setEnabled,"run with XP
1. Create a new Visual bean (SWT)
2. Drop a link on the composite
3. Set the enabled property on the link to be disabled
4. Note: the link is now disabled in the preview
5. Close the editor
6. Re-open the editor
7. Observe: link now shows as enabled
When you run the application the link shows as enabled, but is actually disabled.
If you change the order of the initialization such that the enabled status is set after the text is set it displays correctly.",org.eclipse.swt.widgets.Link
FILE,eclipse-3.1,104987,2005-07-25T06:43:00.000-05:00,view contents missing when switching perspectives,"- start eclipse on a fresh workspace
open in Java perspective
- create a java project and a class Foo.java
- switch to the debug perspective
-> outline view has no contents
-> debug view is empty
-> variables view is empty
debug view
switch to other perspectives","org.eclipse.ui.internal.PartStack
org.eclipse.ui.internal.PartPane"
FILE,eclipse-3.1,104991,2005-07-25T07:07:00.000-05:00,[WorkbenchParts] Cannot reopen closed views,"- new workspace
- close Problems view
- Window > Show View > Problems
-> view is not opened, nothing in the log
open view
break in n20050720-0010",org.eclipse.ui.internal.PartStack
FILE,eclipse-3.1,106725,2005-08-11T04:57:00.000-05:00,assist ] wrong method,"build id build i20050627-1435
create some class with method main.
create another class that extends first class.
type ""main"" in class body, press ctrl-space for content assistant choose there
""main - main method"".
method toString_ is created.","org.eclipse.jdt.internal.codeassist.RelevanceConstants
org.eclipse.jdt.internal.codeassist.CompletionEngine"
FILE,eclipse-3.1,107391,2005-08-18T21:20:00.000-05:00,not work with trailing trail in code style prefix list,"While editing my code style prefix list regarding generating getters and setters I appended a comma at the end.
remove comma start after removing
I found this stacktrace, which will help pinpoint the problem.",org.eclipse.jdt.internal.corext.util.Strings
FILE,eclipse-3.1,108162,2005-08-26T14:27:00.000-05:00,[Progress] [IDE] Deadlock using MoveFilesAndFoldersOperation from a refactoring participant,"move refactoring participant for resources
The following line of code in the participant causes the deadlock:
be from org.eclipse.ui.actions package",org.eclipse.ui.actions.WorkspaceModifyOperation
FILE,eclipse-3.1,108473,2005-08-31T11:22:00.000-05:00,[Performance] The performance of eclipse severely degrades over time,"have same problem on Windows XP
run Tomcat from Eclipse use Sysdeo launcher
After running Tomcat several times, there is a huge slow down in Tomcat:
Almost all the CPU time (more than 80%) is allocated to a thread in Eclipse,
Tomcat itself is not consuming much CPU time (less than 1%) but is not responding correctly at all.
debug mode run tomcat process in debug
stop Tomcat run without debug debug mode
have debug mode restart eclipse",org.eclipse.jdt.internal.debug.ui.JavaDebugOptionsManager
FILE,eclipse-3.1,109104,2005-09-08T17:35:00.000-05:00,Editor save does not write to disk (intermittent),"Several times while using this build, I have hit save in an editor but the buffer contents have not been flushed to disk.
When this happens, the file on disk has the same contents as the previous save, but the ""Save"" action is greyed out and the dirty indicator (*) is gone from the editor tab.
There are no errors in the log file or on the Java console.
not line up with correct code
open same file in Wordpad not match contents in wordpad
dirty editor hit editor flush correct contents to disk
do recent time
1) Did a search and replace in a Java file (Ctrl+F, Replace All)
2) Hit save.
3) Discovered that I replaced with the wrong thing
4) Hit Undo (Ctrl+Z)
5) Did another search and replace with the correct value
6) Hit Save.
The editor now looks correct, but the file on disk still looks like it was in step 2).
I will attach a screen shot in case it helps clarify.",org.eclipse.jface.text.AbstractDocument
FILE,eclipse-3.1,109254,2005-09-11T12:02:00.000-05:00,[Algorithms] Compare causes entire machine to hang when comparing large files,"1. Commit a file of >50k lines.
2. Compare it against a similar local file.
- After a while I give up and press Cancel.
- Eclipse and all other applications are unresponsive
- after a long time(somewhere between 2-20 minutes, I didn't check)
the requester goes away and Eclipse puts up an error dialog that it was unable to compare the files.
compare files
compare execution traces of complicated programs","org.eclipse.compare.rangedifferencer.RangeDifferencer
org.eclipse.compare.rangedifferencer.LinkedRangeDifference
org.eclipse.compare.internal.CompareUIPlugin
org.eclipse.compare.rangedifferencer.RangeDifference"
FILE,eclipse-3.1,110837,2005-09-27T13:16:00.000-05:00,javax.crypto.KeyAgreement.getInstance(String) throws exception in IDE,"A call to KeyAgreement.getInstance(""DiffieHellman"") will throw a
NoSuchAlgorithmException if run from Eclipse but not if it's run directly from the command line.
Here's the code:
import java.security.NoSuchAlgorithmException;
import javax.crypto.KeyAgreement;
throw NoSuchAlgorithmException
run from command line go command line
Running in Eclipse with the same JDK and environment produces this exception:","org.eclipse.jdt.launching.AbstractJavaLaunchConfigurationDelegate
org.eclipse.jdt.internal.launching.JRERuntimeClasspathEntryResolver
org.eclipse.jdt.internal.launching.StandardVMType"
FILE,eclipse-3.1,112137,2005-10-10T16:22:00.000-05:00,ConcurrentModificationException when CTRL+LeftClick on constructor_ call in Java editor,"While coding in Java, I CTRL+clicked on an allocation call to InetSocketAddress, in order to open the sources and read a bit.
However, the call failed, and below you'll find what was printed in the log.
close failed window do after closing work after closing
not find bug have idea get something out_of stack trace
create editor ID org.eclipse.jdt.ui.ClassFileEditor",org.eclipse.jdt.internal.core.SourceMapper
FILE,eclipse-3.1,126938,2006-02-08T12:20:00.000-06:00,Cannot import Installed JREs from preference file,"Below are the contents of the preferences file I am trying to import.
To reproduce, you may need to change the location of the JRE since you probably don't have it installed in the directory I do.
r \ n. r \ n</vmType>
1. Create a new workspace.
2. Import a preference file.
3. Open the preference window and view the JRE's",org.eclipse.jdt.internal.launching.JREPreferenceModifyListener
FILE,eclipse-3.1,133072,2006-03-23T16:58:00.000-06:00,"Cannot launch an ""Eclipse Application"" without the -ws argument","launch Eclipse application do NOT use SWT from 3.2M5 do NOT use SWT under OSX do Eclipse application from 3.2M5 do Eclipse application under OSX get following problems
When running against 1.4.2_09, you get the dreaded ""2006-03-23
load java [ 1053 ] apple AWT java VM on first thread
The application then stops.
When running against 1.5.0_06 you get these messages on startup:
Under 1.5.0_06, the app appears to run, but there is no menu-bar,
dock-icon, or window that shows up.
Basically nothing happens, but
the event thread is running and you have to kill it.
tell Eclipse not force first-thread flags by passing not pass ws carbon flag to application
have ability suppress flag
remove check for ws flag rebuild plugin hack by rebuilding hack by removing
override ws flag want per Launch configuration
Here's the test class we used:
import javax.swing.SwingUtilities;
import org.eclipse.core.runtime.IPlatformRunnable;
implement IPlatformRunnable {
throw Exception { SwingUtilities.invokeLater(new Runnable() {                        public void run() {                                new JFrame(""Fred"")","org.eclipse.pde.internal.ui.IPDEUIConstants
org.eclipse.pde.internal.ui.launcher.LaunchAction"
FILE,eclipse-3.1,142881,2006-05-20T00:02:00.000-05:00,work on non-editable editor,"1. Open Plug-ins View
2. Double-click on any plug-in with a blue/purple icon (e.g. org.eclipse.core.runtime.compatibility.auth.
3. On the Overview page, select an item in the Execution Environments table.
press DEL key
Observe how the selected item gets deleted.","org.eclipse.pde.internal.ui.editor.plugin.ImportPackageSection
org.eclipse.pde.internal.ui.editor.plugin.RequiresSection
org.eclipse.pde.internal.ui.editor.plugin.ExtensionPointsSection
org.eclipse.pde.internal.ui.editor.plugin.ExportPackageSection"
FILE,eclipse-3.1,157944,2006-09-20T01:11:00.000-05:00,JavaAttributeWizard shouldn't open with errors when passed an empty string,"test candidate
1. On the overview page of the plugin editor, make sure the Activator field is empty.
2. Click the Activator link.
The class wizard opens with an error.
fix for bug",org.eclipse.pde.internal.ui.editor.plugin.JavaAttributeWizardPage
FILE,eclipse-3.1,164438,2006-11-14T04:04:00.000-06:00,mistake for plug-ins,"When you import features from the Target platform via the Feature Import wizard, the imported features do not recognized by PDE as features in the workspace.
Upon investigation, it turns out that this is due to the fact that the imported features are signed.
cause confusion between plug-in project structure cause confusion between feature project structure
PDE is treating signed feature projects as plug-in projects.",org.eclipse.pde.internal.ui.wizards.imports.FeatureImportOperation
FILE,eclipse-3.1,300054,2010-01-19T10:12:00.000-06:00,Unexpected 'Save Resource' dialog appears when copying changes from right to left,"1 start with new workspace
2 paste this into the Package Explorer:
void bar() {
4 delete ""System.out.println();"" and save
5 compare the current state with the previous one
6 double-click on 'bar' in the compare editor's upper pane
7 click 'Copy Current Change from Right to Left' button
==> 'Save Resource' dialog appears which is a major interruption of my workflow.
focus compare editor on method",org.eclipse.compare.internal.Utilities
FILE,eclipse-3.1,76534,2004-10-18T22:57:00.000-05:00,Can't perform evaluations inside inner class with constructor_ parameters,"We currently disallow evaluations in inner classes that take parameters in the referenced constructor_.
For example, see the CheckBoxTreeViewer that's created in BreakpointsView#createViewer(...).
allow evaluations in kinds",org.eclipse.jdt.internal.debug.eval.ast.engine.SourceBasedSourceGenerator
FILE,eclipse-3.1,76869,2004-10-23T01:13:00.000-05:00,3.1M2: update configurator doesn't find feature.xml because of .DS_Store,"I added the Web Tools into plugins/ and into features/.
After starting the SDK, the web tools aren't available.
start with following entries
find feature.xml in directory
scan Target platform
require plug-in org.eclipse.wst.server.core_0.0.0
find feature.xml in directory
have problems
find in mac OS x folders",org.eclipse.update.internal.configurator.SiteEntry
FILE,eclipse-3.1,77234,2004-10-28T15:41:00.000-05:00,not see inherited method,"1 Create a detail formatter for the type JavaExceptionBreakpoint.
2 Set the contents to ""getTypeName()""
3 Debug to a breakpoint with a JavaExceptionBreakpoint in the variables view.
I'm debugging RemoveBreakpointAction and I delete an exception breakpoint to see
this.
4 Select the JavaExceptionBreakpoint.
I get the following in the details pane:
extend JavaBreakpoint declare getTypeName() on JavaBreakpoint",org.eclipse.jdt.internal.debug.ui.JavaDetailFormattersManager
FILE,eclipse-3.1,77239,2004-10-28T16:56:00.000-05:00,navigate ill-formed stack,"navigate stack trace
For example, pasting the 
following trace into the console has these problems:
(1 partial lines such as ""e.run(JavaWatchExpressionDelegate.java:144)""
results in a hyperlink that can't find source for ""e""
(2 removing the newline character to create one complete line does not
recalcualte the hyperlinks","org.eclipse.jdt.internal.debug.ui.JavaDebugImages
org.eclipse.jdt.internal.debug.ui.console.JavaStackTracePageParticipant
org.eclipse.jdt.internal.debug.ui.console.FormatStackTraceActionDelegate"
FILE,eclipse-3.1,78160,2004-11-09T08:03:00.000-06:00,classfile for problem type,"When compiling the following testcase, the problem classfile issued has an invalid signature in it:
extend y < y <
extend y < y <
not extend y < y < implement y < y <
not specify wildcard
have inconsistent hierarchy in class have illegal signature v have class",org.eclipse.jdt.internal.compiler.lookup.ClassScope
FILE,eclipse-3.1,78460,2004-11-11T18:37:00.000-06:00,[Contributions] [ViewMgmt] Console actions leaking after each launch,"For each launch that terminates and is removed, two console actions are leaked
(which also keep a fair bit of the previous launch session in memory).
Ran a program to termination.
Removed the program via the Debug view
ConsoleTerminateActionDelegate and ConsoleRemoveAllTerminatedActionDelegate leaked per launch.
keep reference in memory
not free up ....","org.eclipse.ui.console.TextConsolePage
org.eclipse.debug.internal.ui.actions.AbstractDebugActionDelegate
org.eclipse.ui.internal.EditorPluginAction
org.eclipse.ui.internal.ViewPluginAction
org.eclipse.ui.internal.ViewerActionBuilder
org.eclipse.ui.internal.PluginActionBuilder
org.eclipse.ui.internal.EditorActionBuilder"
FILE,eclipse-3.1,78740,2004-11-16T10:57:00.000-06:00,IDOMType.getFlags() fails to represent interface flags correctly.,"This code demonstrates that calling getComment on an IDOMType will change the flags from correctly encoding the type as being interface to incorrectly encoding it (because during becomeDetailed() that information is lost):
package org.example.jdom;
import org.eclipse.core.runtime.IPlatformRunnable;
import org.eclipse.jdt.core.Flags;
import org.eclipse.jdt.core.jdom.DOMFactory;
import org.eclipse.jdt.core.jdom.IDOMCompilationUnit;
import org.eclipse.jdt.core.jdom.IDOMType;
implement IPlatformRunnable
return new integer
break JavaEcoreBuilder",org.eclipse.jdt.internal.compiler.DocumentElementParser
FILE,eclipse-3.1,79318,2004-11-23T12:49:00.000-06:00,[ccp] replacing directory with a file of the same name containing this very file will delete file and directory,"There is a directory/folder named ""bar"" somewhere in the workspace.
This directory contains a file also called ""bar"".
When trying to move the file ""bar"" one level up, Eclipse will show a dialog stating that there already is a file named ""bar"" in this directory and asking the user if he really wants to overwrite it.
If the user clicks ""Yes"" Eclipse will run into an exception and will show this error dialog: ""An exception has been caught while processing the refactoring
There are two buttons: ""Undo"" and ""Abort"".
No matter what button is clicked, both the directory ""bar"" and the file ""bar"" are lost.
Presumably Eclipse/JDT deletes/overwrites the directory ""bar"" before the file
""bar"" could be read.
not allow user move file overwrite containing directory
make backup copy before replacing make backup copy of file replace directory with file
affect Package explorer
move files",org.eclipse.jdt.internal.corext.refactoring.reorg.OverwriteHelper
FILE,eclipse-3.1,79793,2004-11-30T12:22:00.000-06:00,Deadlock in JavaWorkingSetUpdater,"* Started workspace
* Pressed CTRL-Shift-T to open a type
* Progress dialog appeared
* pressed cancel
* Pressed CTRL-Shift-T again
> workspace deadlocked
attach thread dump",org.eclipse.jdt.internal.ui.workingsets.JavaWorkingSetUpdater
FILE,eclipse-3.1,79957,2004-12-02T00:47:00.000-06:00,[Viewers] NPE changing input usingTableViewer and virtual,"use latest code with private virtual manager class use latest code for Table viewer
i've straight forward code ... 
<code>
Table table=new Table(shell,SWT.VIRTUAL);
TableViewer tv=new TableViewer(table);
tv.setContentProvider(new NetworkContentProvider());
tv.setLabelProvider(new NetworkLabelProvider());
tv.setInput(model);
.
reset model input
Same code works fine without the SWT.VIRTUAL style bit,but when VIRTUAL is set
it throws a null pointer exception...",org.eclipse.jface.viewers.TableViewer
FILE,eclipse-3.1,80387,2004-12-07T12:10:00.000-06:00,NPE opening compare editor,"I got this while opening a compare editor from the synch view.
invoke code from plug-in",org.eclipse.compare.CompareEditorInput
FILE,eclipse-3.1,81045,2004-12-14T20:13:00.000-06:00,ClassNotLoadedException when trying to change a value,"void main(String[] args) {
Debug to the breakpoint.
Right-click on the 'inner' variable > change value ...
A ClassNotLoadedException dialog appears.","org.eclipse.jdt.internal.debug.ui.actions.JavaVariableValueEditor
org.eclipse.jdt.internal.debug.eval.ast.engine.ASTEvaluationEngine
org.eclipse.jdt.internal.debug.core.model.JDILocalVariable"
FILE,eclipse-3.1,81333,2004-12-15T16:51:00.000-06:00,"[GlobalActions] ""Open Resource"" dialog doesn't filters out class files?","1 Press ctrl+shift+r to open the ""Open Resource"" dialog.
2 Type the name of a class file in the workspace (for example, ""Hello.class"").
3 The dialog does not show the class file, even though it exists.
do with simple hello world program do in case do in default package
Hello.java and Hello.class were in the same directory, but the dialog
could only find Hello.java.","org.eclipse.ui.internal.ide.dialogs.OpenResourceDialog
org.eclipse.ui.views.navigator.GotoResourceDialog
org.eclipse.ui.dialogs.ResourceListSelectionDialog"
FILE,eclipse-3.1,82712,2005-01-12T15:54:00.000-06:00,[1.5] Code assist does not show method parameters from static imports,"Test:
import static java.lang.Math.
void t() { abs",org.eclipse.jdt.internal.codeassist.CompletionEngine
FILE,eclipse-3.1,83216,2005-01-19T13:19:00.000-06:00,change in new Working set,"I20050118-1015
Navigator view menu>Select Working Set>New
Select a working set type
Next
Back
Notice the change in space around the ""Working set type"" label
change Dialog font to larger size","org.eclipse.jdt.internal.compiler.ast.ParameterizedQualifiedTypeReference
org.eclipse.jdt.internal.compiler.ast.ParameterizedSingleTypeReference
org.eclipse.jdt.internal.compiler.ast.QualifiedAllocationExpression
org.eclipse.ui.internal.WorkbenchPage
org.eclipse.ui.internal.ide.dialogs.ResourceWorkingSetPage"
FILE,eclipse-3.1,83489,2005-01-22T17:33:00.000-06:00,[select] Code select returns IType instead of ITypeParameter on method parameters types,"use HEAD
Consider following test case:
When I select ""T"" in method declaration, selection engine returns an IType
""Test"" instead of expected ITypeParameter ""T"".",org.eclipse.jdt.internal.codeassist.SelectionEngine
FILE,eclipse-3.1,84194,2005-02-01T18:05:00.000-06:00,[content assist] Code assist in import statements insert at the end,"Open any Java file that contains > 1 import statements.
Let's say the first import statement reads:
import org.eclipse.core.runtime.
delete the '*;' from the end and try to use code assist to insert
IRunnableWithProgress for example.
You will see that upon pressing Enter to select, the text gets inserted several lines down under all the import statements.
the cursor is now in a random position also.","org.eclipse.jdt.internal.ui.text.java.JavaTypeCompletionProposal
org.eclipse.jdt.internal.ui.text.java.ExperimentalResultCollector"
FILE,eclipse-3.1,84557,2005-02-07T07:06:00.000-06:00,Table item does not appear in table,"run i20050202-0800 linux gtk
ctrl esc E brings up the handy switch to editor window, but unfortuantley the list is empty :(","org.eclipse.swt.widgets.TableColumn
org.eclipse.swt.widgets.TreeColumn"
FILE,eclipse-3.1,84652,2005-02-08T03:05:00.000-06:00,Build site fails with non std. directory structures,"I have a workspace with 3 plugin projects, one feature project and one site project.
all projects are physically not in the workspace but at different positions on the harddrive.
the layout is something like this:
+dir1
+ plugin 1
+ plugin 2
+ feature
+ site
+dir2
+ plugin 3
the workspace looks something like this
+plugin1
+plugin2
+plugin3
+feature
+site
plugin1 and plugin2 both make use of plugin3.
if i try to build the site i get classnotfound/import not found errors from the compiler for each reference to plugin3.
strangely a can generate a build.xml file from the feature and build the feature with build.update.jar without getting errors.
place plugin3 right build site expected
have idea be idea find bugs with linked sourcefolders search in bugzilla database",org.eclipse.pde.internal.build.builder.ClasspathComputer3_0
FILE,eclipse-3.1,84724,2005-02-08T13:41:00.000-06:00,[1.5][search] fails to find call sites for varargs constructor_s,"The search engine fails to find the call to the varargs constructor_ in the example below.
Simply highlight the constructor_'s name and invoke ""References"" -
> ""Workspace"" from the Java editor context menu; no occurrences will be found.
call Cell.Cell(String...)",org.eclipse.jdt.internal.core.search.matching.ConstructorLocator
FILE,eclipse-3.1,84881,2005-02-10T07:19:00.000-06:00,[WorkbenchLauncher] Splash screen receives focus after task switch (instead of workspace launcher dialog),"When starting Eclipse 3.1, the splash screen and the ""select a workspace"" dialog
(workspace launcher) is shown.
If you then switch to another task and switch
back to eclipse using ALT+Tab (MS Windows), the Splash-Screen receives the focus
while the ""Select a workspace"" dialog stays in front but is inactive.
select workspace continue workspace not have mouse
I've not found any workaround to
this problem (closing the splash-screen via ALT+F4 removes the task from the
task bar, while the dialog stays in front without focus).","org.eclipse.ui.internal.ide.IDEApplication
org.eclipse.ui.internal.ide.ChooseWorkspaceDialog
org.eclipse.ui.internal.Workbench"
FILE,eclipse-3.1,85696,2005-02-17T09:06:00.000-06:00,"Spurious update message repeats infinitely, unable to use Eclipse","leave i20050215-2300
When I came in this morning it said new
updates were available, would I like to install them?
I said yes, and it opened
the Search Results window.
However that window does not list any updates.
It
says No updates for the currently installed features(s)found, try again later.
find )
The option to Show the latest
version of a feature only is checked, and the option to Filter patches included
in other patches is unchecked.
change options not have effect
All the buttons were grayed out except Deselect All (which was wierd
because there was nothing to deselect) and Cancel.
When I clicked Cancel it gave me the Automatic Updates dialog again.
I selected Yes again, got the empty Search Results dialog again.
So the next time I said No.
I got the first ""New updates are
available..."" dialog again, yes/no.
Either way leads to infinite repetition.
I'm
unable to use Eclipse or even see the Workbench window currently.",org.eclipse.update.internal.scheduler.SchedulerStartup
FILE,eclipse-3.1,85734,2005-02-17T12:28:00.000-06:00,Debug view flickers excessively,"linux 2.6.10
The debug view flickers excessively when debugging.
In particular, I have set a
breakpoint on ""Runtime.exec(...)"" and started a debugging session on Eclipse.","org.eclipse.debug.internal.ui.views.RemoteTreeViewer
org.eclipse.debug.internal.ui.views.launch.LaunchViewer
org.eclipse.debug.internal.ui.views.launch.LaunchViewEventHandler
org.eclipse.debug.internal.ui.views.RemoteTreeContentManager"
FILE,eclipse-3.1,86000,2005-02-21T14:47:00.000-06:00,ImageLoader Save - produces invalid JPEG images,"The ImageLoader Save function appears to be producing bad JPG images.
Simple test case below loads
 PNG Files and Saves them as JPEG.
Many files were tested and the majority 
 did produced the proper JPG images as expected.
contain files not save files to JPEG
package com.ibm.test.image;
import org.eclipse.swt.
try { for { string filein = dir + files [ i ] +","org.eclipse.ui.internal.WorkbenchIntroManager
org.eclipse.swt.internal.image.JPEGFileFormat"
FILE,eclipse-3.1,86725,2005-02-25T18:50:00.000-06:00,var on windows,"think on WinOS
see LaunchManager.getEnvironment for gory details
investigate through bug report
write in message news:cvo808
work through Eclipse ( work via command line
have my_env_var declare at system level
make system vars
note for others
not get set not get environment variables
do work
write in message
> >> For some reason, when I run my build script in M5, it's not recognizing
> >> this line:
> >> So later on when I use something like:
> >> it's not able to resolve the variable system.my_env_var.
run in 3.0.1 run via command line
change ANT version eclipse see behavior
run with jdk 1.4.2_06","org.eclipse.debug.internal.core.EnvironmentVariableResolver
org.eclipse.debug.internal.core.LaunchManager
org.eclipse.debug.ui.EnvironmentTab
org.eclipse.debug.core.ILaunchManager"
FILE,eclipse-3.1,87665,2005-03-10T11:38:00.000-06:00,Clicking on x on performance page opens details with no errors,"Take a look at:
Scroll down to performance.OpenJavaEditorStressTest#testOpenJavaEditor1()""
Click on the red x ==> details show up all green.
like bug regarding handling","org.eclipse.swt.printing.PrintDialog
org.eclipse.swt.widgets.MessageBox"
FILE,eclipse-3.1,87683,2005-03-10T13:36:00.000-06:00,show unshortened text,"On the menu bar from the main Workbench panel, click Window > Preferences > Java
> Editor
Select Hovers tab in the right panel
Enable all the text hovers, an error message will display.
The error message is
truncated, no matter how big you enlarge the screen.
see screenshots",org.eclipse.jface.dialogs.DialogMessageArea
FILE,eclipse-3.1,88328,2005-03-17T09:42:00.000-06:00,"bogus ""application o.e.ui.ide.workbench will fail to run"" error when validating plug-in set","launch configuration
If I change it to be [No
Application - Headless Mode], and make sure only the plug-ins I will actually need are selected in the Plug-ins tab (that excludes org.eclipse.ui.ide), and fire ""Validate Plug-in Set"", the validation will return an error about not being able to run org.eclipse.ui.ide.workbench application, because its plug-in",org.eclipse.pde.internal.ui.launcher.AdvancedLauncherTab
FILE,eclipse-3.1,89259,2005-03-28T13:50:00.000-06:00,[WorkbenchParts] assertion failed when opening Compare editor for MANIFEST.MF,"Opening compare editor for MANIFEST.MF files fails with the following AssertionFailedException.
- select two random MANIFEST.MF files
- compare with each other
The same happens when comparing with CVS resource history.
create editor ID org.eclipse.compare.CompareEditor","org.eclipse.ui.IEditorInput
org.eclipse.ui.internal.PartTester"
FILE,eclipse-3.1,89682,2005-03-30T15:48:00.000-06:00,inconsistent with defnition inconsistent for location props,"* In the fowllowing buildfile, if I put the cursor in a reference to a property, like ""destBase"", all occurrences are found.
If I place the cursor on the definition of ""destBase"" (i.e. in the property def itself), no occurrences are found.
build file check out for simplifying","org.eclipse.ant.internal.ui.model.IAntModelConstants
org.eclipse.ant.internal.ui.model.AntPropertyNode
org.eclipse.jdt.internal.ui.packageview.PackageExplorerActionGroup"
FILE,eclipse-3.1,89883,2005-03-31T15:22:00.000-06:00,Exception triggered by Memory View,"Got the following exception when starting the workbench and the Memory View is 
visible.
get selection from Memory view
attempt by view org.eclipse.debug.ui.MemoryView
attempt by view
attempt by view org.eclipse.debug.ui.MemoryView","org.eclipse.debug.internal.ui.views.memory.RenderingViewPane
org.eclipse.debug.internal.ui.views.memory.MemoryBlocksTreeViewPane
org.eclipse.debug.internal.ui.views.memory.NewMemoryViewAction"
FILE,eclipse-3.1,90782,2005-04-08T07:34:00.000-05:00,DBCS3.1: Internal error occurred retrieving targets,"1 select a Ant build file and select Run As > Ant Build...
Result
Internal error ocurred retrieving target was shown in .
run in 3.0.2","org.eclipse.ant.internal.ui.AntUtil
org.eclipse.jdt.internal.ui.wizards.buildpaths.UserLibraryMarkerResolutionGenerator"
FILE,eclipse-3.1,91098,2005-04-12T06:07:00.000-05:00,The Mark Occurrences feature does not mark all occurrences,"The precise test case is the following:
Put the cursor on String or String[].
All occurrences of String get highlighted.
Now put the cursor on String[][].
No occurrence of String is highlighted.
remove square brackets
use 3.1M6",org.eclipse.jdt.core.dom.ASTConverter
FILE,eclipse-3.1,92063,2005-04-20T09:30:00.000-05:00,Editor closed while replacing corresponding file with HEAD contents,"have Java editor with changes
In order to do so, I selected unit in package explorer, and did ""replace with HEAD"" action.
The editor got closed under me automatically when the replace occurred.",org.eclipse.team.internal.ccvs.ui.operations.ReplaceOperation
FILE,eclipse-3.1,92451,2005-04-22T16:36:00.000-05:00,assist failure cast + arrays,"Code assist fails in the following (self-contained) class (see comments for line of error)
work new testelements.toArray on line
fail on line",org.eclipse.jdt.internal.codeassist.complete.CompletionParser
FILE,eclipse-3.1,93727,2005-05-04T17:43:00.000-05:00,Code Formatter fails with Method Parameter Annotations,"have methods with parameter annotations
It fails silently, and I don't see an error in
<Workspace>/.
Example:
import org.drools.semantics.annotation.DroolsParameter;",org.eclipse.jdt.internal.formatter.CodeFormatterVisitor
FILE,eclipse-3.1,94216,2005-05-09T20:04:00.000-05:00,not work for generic types,"implement IGeneric<T>
Try to do 'open declaring type' or 'open concrete type' for 'gen' at the breakpoint, nothing happens.","org.eclipse.jdt.internal.debug.ui.actions.OpenVariableDeclaredTypeAction
org.eclipse.jdt.internal.debug.ui.actions.OpenVariableConcreteTypeAction"
FILE,eclipse-3.1,94465,2005-05-10T14:33:00.000-05:00,Java Core Dump where modifying value in the Variables View.,"Test case:
have string array
1. In the variables, expand the array, and then expand the first element, [0]
=""abc"".
2. Select the ""value=char[3]"" field.
RMC->Change Value.
3. In the Change Primitive Value dialog, type in a new string value.
4. Click ok and it will result in a java dump.
get following error in console
write to d
have Processed exception Signal","org.eclipse.jdt.internal.debug.ui.JDIModelPresentation
org.eclipse.jdt.internal.debug.ui.actions.JavaObjectValueEditor
org.eclipse.jdt.internal.debug.ui.actions.ActionMessages"
FILE,eclipse-3.1,94769,2005-05-11T11:58:00.000-05:00,Step over (macrodef) resumes Ant debugger,"* in the attached script, I placed a breakpoint on the ""generate all"" target
* stepped over, over, and into ""<process destdir=""${destBase}\exercise1.1"" symbols=""ex1""/>"", and ""into"" again to get into the marcro def
* stepped thur the macrodef using ""step over""
* when I stepped over the last line of the macrodef (""preprocess""), the ant script  simply resumed to completion
build file check out for simplifying",org.eclipse.ant.internal.ui.antsupport.logger.util.AntDebugState
FILE,eclipse-3.1,94852,2005-05-11T17:17:00.000-05:00,activate view menu button,"Give focus to a view menu button.
For example, in the Package Explorer view, 
give focus to the tree, and type shift+tab, shift+tab.
The view menu (little 
down-pointing triangle) button should have focus (it looks a bit raised).
have focus give hint to users activate button by pressing press space
Space does 
nothing when pressed in this button.
I realize that pressing Ctrl+F10 does drop down the menu, therefore typing 
space is not technically needed for accessibility, however it is really 
strange that here is a button (in a toolbar) that does nothing when ""pressed"" 
using the keyboard space key.
(Note 
that all of our other toolbar buttons do activate when they have focus and 
space is pressed - including the very similar ""Open Perspective"" menu toolbar 
button).",org.eclipse.ui.internal.presentations.defaultpresentation.DefaultTabFolder
FILE,eclipse-3.1,95152,2005-05-13T12:14:00.000-05:00,[search] F3 can't find synthetic constructor_,"1) Add org.eclipse.debug.ui to the search path (i.e., by clicking ""Add to Java
Search"" in the plugins view.
2) Open type on ""ProcessConsole"" (class file with source attached)
3) Go to line 483:
4) Highlight the InputReadJob constructor_ and hit F3.
open new class file editor position at top
have constructor _
Clicking this entry in the outline view does not jump to the constructor_ in the editor.
The mapping of class file to source is not handling the synthetic addition of the enclosing class by the compiler.
This breaks any kind of navigation to the corresponding constructor_ in the source attachment.","org.eclipse.ant.internal.ui.views.AntViewDropAdapter
org.eclipse.ant.internal.ui.launchConfigurations.AntLaunchShortcut
org.eclipse.ant.internal.ui.AntUtil
org.eclipse.jdt.internal.core.search.matching.ConstructorLocator
org.eclipse.jdt.internal.core.search.indexing.BinaryIndexer
org.eclipse.jdt.internal.core.index.DiskIndex
org.eclipse.jdt.internal.core.search.matching.ConstructorPattern"
FILE,eclipse-3.1,95490,2005-05-16T23:39:00.000-05:00,String substitution variables are lost during import,"Run/Debug string substitution variables are lost during import of preferences.
To reproduce import preferences from the attached file.
do work in m5",org.eclipse.core.internal.variables.StringVariableManager
FILE,eclipse-3.1,95505,2005-05-17T02:56:00.000-05:00,not use code completion,"It was very convinient in Eclipse that when it already knows type and I write
""new "" and press Ctrl+Space, it shows this type.
For example:
have look in m7 write several first letters write look
return old behaviour",org.eclipse.jdt.internal.codeassist.CompletionEngine
FILE,eclipse-3.1,95580,2005-05-17T11:34:00.000-05:00,have enum constants have enum type,"Using M7, this would lead to an IllegalArgumentException.
Try to add a field to an enum type that has enum constants.
attach test case attach patch",org.eclipse.jdt.internal.core.CreateFieldOperation
FILE,eclipse-3.1,95902,2005-05-19T03:20:00.000-05:00,PDE leaves manifest.mf out of sync after import,"I imported the binary plug-ins from N20050519-0010 into my existing workpspace which already contained older versions for those plug-ins and used 'File' search to find a plug-in ID.
This did not work: search reported a list of all the imported manifest.mf files being out-of-sync with the workspace, see attached picture.
touch mfs",org.eclipse.pde.internal.core.PDEPluginConverter
FILE,eclipse-3.1,95945,2005-05-19T10:10:00.000-05:00,[Navigator ] Opening compare editor switches to Java editor,"1 Open a .
java, .
xml etc. file that has been modified since last cvs checkout
in resource/java perspective.
2 Synchronize  entire project -> switch to synchronize perspective
3 Double click a changed file that is NOT currently open in perspective - you
get a side-by-side compare view of the two files (good).
4 Double click the changed file that was opened in step 1.
The side-by-side
perspective is displayed for a second, and then the display switches to the
single file view of this file instead of staying in the compare view.
You have to close the file in the perspective in order for the comparison to
keep focus which is really annoying when you want to see the diffs between local
and repository versions on a file you are working on.
not happen in 3.1M5 introduce bug in m6 introduce bug in m7",org.eclipse.ui.views.navigator.ResourceNavigator
FILE,eclipse-3.1,95958,2005-05-19T11:43:00.000-05:00,Opening a closed project causes exception in Breakpoints View,"1. Create multiple projects and files.
2. Add some breakpoints in different projects
3. Show breakpoints with Project Group.
4. Close one of the projects.
5. Reopen the closed project.
Result in the following exception and Breakpoints View can no longer show breakpoint unless the workbench is restarted:
execute runnable
ru n(BreakpointsViewEventHandler.java:91)",org.eclipse.debug.internal.ui.views.breakpoints.BreakpointsViewEventHandler
FILE,eclipse-3.1,96294,2005-05-23T07:11:00.000-05:00,double clicking on a stack frame / hitting a breakpoint does not reveal the editor,"Hitting a breakpoint usually takes me to the corresponding location in Java
source.
Also, selecting a stack frame with the mouse will reveal the respectiev
location.
not work in n20050523 work in n20050520",org.eclipse.jdt.internal.debug.ui.JDIModelPresentation
FILE,eclipse-3.1,96440,2005-05-24T11:11:00.000-05:00,Tables laying out 3 times when trying to determine sizes,"layout table
return width in m6 return width of table return smaller value make in Table layout
STEPS
1 Put a breakpoint in the JFace TableLayout class
2 Launch a self hosted workspace
3 Open Preferences-> Ant Runtime
4 You will see a client area size of about 81
5 Do the same in M6 - it will be about 320 or so.",org.eclipse.jface.preference.PreferencePage
FILE,eclipse-3.1,96489,2005-05-24T14:40:00.000-05:00,[Presentations] (regression) Standalone view without title has no border,"build n20050523
- change the browser example's BrowserPerspectiveFactory to have the following instead of the regular addView layout.addStandaloneView(BrowserApp.BROWSER_VIEW_ID, false,
- run the example, and show the history view
- the history view (a regular view) has a border, but the standalone view does not","org.eclipse.ui.presentations.WorkbenchPresentationFactory
org.eclipse.ui.internal.presentations.defaultpresentation.EmptyTabFolder"
FILE,eclipse-3.1,96859,2005-05-26T15:37:00.000-05:00,remove validation at runtime remove validation of default target,"Hi,
having the build.xml file like
<?xml version=""1.0"" encoding=""UTF-8"" ?>
<project name=""namespace-test"" default=""hip"" basedir=""."">
	<target name=""hello"">
	</target>
</project>
1 I get no error saying that there is no target ""hip""
2 I can run the file with ""Run as->Ant build"" adn get a failure saying
not exist in project
3 Even if I choose the ""hello"" target in ""Run as->Ant build ..."" I get an error
since still the ""hip"" target is executed, ignoring my setting!
4 If I remove the default=""hip"", I can run the ""hello"" target.
consult ant","org.eclipse.ant.internal.ui.antsupport.InternalAntRunner
org.eclipse.ant.internal.core.ant.InternalAntRunner"
FILE,eclipse-3.1,97018,2005-05-27T11:47:00.000-05:00,PDE generates to strict access restrictions,"- download EMF using the update manager (don't know if this is necessary)
- observe: the install plug-in has a manifest file which only contains one line:
- it also contains a plugin.xml file which exports the following entries:
This result in the following access restriction for clients using emf.core
forbid reference
As a result all usages to emf core classes even those in API package are flagged as an error.","org.eclipse.pde.internal.core.MinimalState
org.eclipse.pde.internal.core.ClasspathUtilCore"
FILE,eclipse-3.1,97085,2005-05-27T17:10:00.000-05:00,not propose <package>,"With J2SE 5.0 compliance enabled, type the following into a new Java file:
import static java
Code assist will propose all subpackages of java, each followed by "".
*;"", which 
results in a compile-time error if accepted.",org.eclipse.jdt.internal.codeassist.CompletionEngine
FILE,eclipse-3.1,97608,2005-05-31T13:36:00.000-05:00,"Workspace save fails if marker contains > 65,536 byte message","If you create a marker with a String property containing more than 65,536 bytes, it causes workspace save to fail.
restore workspace in next session
see bug for details",org.eclipse.core.internal.resources.MarkerInfo
FILE,eclipse-3.1,97818,2005-06-01T08:18:00.000-05:00,entry in .cvsignore is not ignored on update/sync,"in our projects we have the following structure
an entry in .
cvsignore is target.
M7 and prior ignored m7 when syncing/updating, rc1 does not.
occur in projects",org.eclipse.team.core.variants.AbstractResourceVariantTree
FILE,eclipse-3.1,98147,2005-06-02T13:09:00.000-05:00,Variables View does not show all children if same instance is expanded twice,"- Debug the class below with the breakpoint where indicated.
- Expand t1 in the Variables view -> expands fine and shows fID and fName.
- Expand t2 -> only child fID is shown
void main(String[] args) {",org.eclipse.debug.internal.ui.views.RemoteTreeViewer
FILE,eclipse-3.1,98272,2005-06-03T10:04:00.000-05:00,work for terminate,"not use due_to change
With a project set up to run, first have the debug view open.
Run the application in your project.
You will see in the debug view:
- <Project> [Java Application]
-<path>/j2re1.4.2/bin/javaw.exe(Date started)
This is the issue- in 3.0 and earlier, the javaw.exe line if the last run application would AUTOMATICALLY get selected.
This in turn would enable the terminate button in this view (and thus the key binding).
In 3.1 however, there is no selection made, so the terminate button is not enabled (since there is no context of what you want to terminate) and thus the key binding won't work.
open view make selection",org.eclipse.debug.internal.ui.views.launch.LaunchViewEventHandler
FILE,eclipse-3.1,98604,2005-06-06T18:08:00.000-05:00,change Package explorer,"See bug 98602 for initial steps to reproduce.
After doing that, I selected PDE Tools > Update Classpath.
It updated the classpath, all right, but in the process changed the source display in the affected project (only!)
from flat to hierarchical.
The src folder moved from the first displayed item in the project to the fourth (after the META-INF folder.
see bug
not find way change explorer options be explorer options for individual project
delete project check out project",org.eclipse.pde.internal.core.ClasspathComputer
FILE,eclipse-3.1,98740,2005-06-07T13:25:00.000-05:00,Container attempts to refresh children on project that is not open,"Take an existing simple project on disk and import the project into the workspace by performing a simple create with code like:
Do not open the project with the project.open() API.
Now create a project either by API or UI and open it.
Or simply switch to the
Java perspective.
A background refresh job has now been started for the closed project, but it never finishes and is stuck in an infinite loop.
be in class org.eclipse.core.internal.resources.Container.
The members() method is excuting if (info.isSet(ICoreConstants.M_CHILDREN_UNKNOWN))
workspace.refreshManager.refresh(this);
not know members because projects
Both the AliasManager and the Java
Perspective are calling members on the IProject.
override method in Project not refresh for closed projects
Our particular use case is that we are loading existing Java projects on disk by performing a create, but never an open.
On the next UI gesture, we get refresh infinite loops, one for each closed project.
want projects in workspace
work in Eclipse","org.eclipse.core.internal.resources.Container
org.eclipse.core.internal.resources.Resource"
FILE,eclipse-3.1,99282,2005-06-09T19:46:00.000-05:00,not initialize switch method in thread safe way,"initialize enum/switch table initialize synthetic method place initialization in static initializer
For example, the following program should print ""1999"" 40 times
(once from each thread).
But on my machine, it prints ""default"" 22 times & 1999
18 times.
package com.bea;","org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding
org.eclipse.jdt.internal.compiler.codegen.CodeStream"
FILE,eclipse-3.1,99355,2005-06-10T09:48:00.000-05:00,extract method trips,"if you extract method where indicated below.
you will see that the extracted method declares its paramater with too many final modifiers:
t get()
return new GenericContainer(outerContainer)
void method()
t get()
return new GenericContainer(outerContainer)
void method()
notice the 3 final modifiers in the extractedMethod signature.",org.eclipse.jdt.core.dom.ASTConverter
FILE,eclipse-3.1,99629,2005-06-13T09:17:00.000-05:00,enter expression in change variable value dialog,"I was entering an expression to change the value of an array of doubles, and
received an exception after typing a "".""
in a double, like ""new double[] {1.2
3 ""
execute runnable
execute runnable",org.eclipse.jdt.internal.codeassist.CompletionEngine
FILE,eclipse-3.1,99693,2005-06-13T11:29:00.000-05:00,Invalid stack frames during display,"Debug the following method to a breakpoint:
void doGenerics() {
Select everything between start eval and end eval comments and ctrl-shift-d.
Watching the Variables View I see a lot of invalid stack frames.
They are not destructive (and nothing is logged).
not request and/or cancelling requests","org.eclipse.debug.internal.ui.views.variables.VariablesViewEventHandler
org.eclipse.debug.internal.ui.views.expression.ExpressionViewEventHandler"
CLASS,openjpa-2.2.0,OPENJPA-2227,2012-07-09T14:24:05.000-05:00,OpenJPA doesn't find custom SequenceGenerators,"use custom SequenceGenerator within enterprise application use openJPA
When defining a custom Sequence a ClassNotFoundException (for the Sequence class) will be thrown when trying to insert data into the database.
ExampleConfiguration:
{code}
@Entity
@SequenceGenerator(name=""MySequence"", sequenceName=""org.apache.openjpa.generator.UIDGenerator()"")
public class Customer implements Serializable {
    @Id
    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator=""MySequence"")
    private long id;
{code}
The example will produce the stacktrace attached.
It seems that the wrong class loader is used to instantiate the custom sequence class.
With JavaSE (JUnit) all is working fine, but after deploying into WAS the Exception will occur.
think within method SequenceMetaData.instantiate(Classloader envLoader) use method with parameter mustexist use method instead_of pure Class.forName() call
need for method call",openjpa-kernel.src.main.java.org.apache.openjpa.meta.SequenceMetaData
CLASS,openjpa-2.2.0,OPENJPA-2247,2012-08-03T10:29:58.000-05:00,JoinColumn annotation is ignored when mapping a unidirectional owned OneToOne that is in a SecondaryTable,"The runtime incorrectly ignores @JoinColumn.
name when mapping a unidirectional owned OneToOne that is in a SecondaryTable.
run with persistence.xml set persistence.xml to 2.0
For example:
pchild child
The column ""CHILD_REF"" will be ignored and the runtime will look for the fk in non-existent column ParentSecondaryTable.CHILD_IDCHILD.",openjpa-jdbc.src.main.java.org.apache.openjpa.jdbc.meta.MappingRepository
CLASS,solr-4.4.0,SOLR-5188,2013-08-26T16:53:30.000-05:00,not work in branch_4x not work in trunk,"not work on branch_4x not work on trunk
The /solr/admin/info/logging handler response has an error tag saying Logging Not Initialized.
The problem can be seen by starting the example in the latest checkout from branch_4x or trunk.","solr.core.src.java.org.apache.solr.logging.LogWatcher
solr.core.src.java.org.apache.solr.core.CoreContainer
solr.core.src.java.org.apache.solr.core.ConfigSolr"
CLASS,solr-4.4.0,SOLR-5295,2013-10-02T00:09:02.000-05:00,The createshard collection API creates maxShardsPerNode number of replicas if replicationFactor is not specified,"report by Brett hoerner report on solr-user
set maxShardsPerNode
I've just created a very simple test collection on 3 machines where I set maxShardsPerNode at collection creation time to 1, and
I made 3 shards.
want 4th shard create because cluster
have shard per node
not require more hardware
So I try again -- I create a collection with 3 shards and set maxShardsPerNode to 1000 (just as a silly test).
Now I add shard4 and it immediately tries to add 1000 replicas of shard4...",solr.core.src.java.org.apache.solr.cloud.OverseerCollectionProcessor
METHOD,mahout-0.4,MAHOUT-569,2010-12-28T14:27:05.000-06:00,Category names with spaces cause NullPointerException in ConfusionMatrix when testing classifier,"If you try to perform classification using categories with spaces in their names then the classifier will happily train and create a model for you, but when you go to test the model you get:
thread main java.lang.NullPointerException","org.apache.mahout.classifier.ConfusionMatrix:ConfusionMatrix(Collection<String>, String)
org.apache.mahout.classifier.ConfusionMatrix:getAccuracy(String)
org.apache.mahout.classifier.ConfusionMatrix:merge(ConfusionMatrix)
org.apache.mahout.classifier.ConfusionMatrix:getLabels()
org.apache.mahout.classifier.ConfusionMatrix:getCount(String, String)
org.apache.mahout.classifier.ConfusionMatrix:putCount(String, String, int)"
METHOD,mahout-0.4,MAHOUT-625,2011-03-11T09:24:20.000-06:00,Some of generated patterns have support higher than in reality,"have incorrect support
The returned support is slightly higher than the true one.
attach test have bug
Test is using data (retail) found here: http://fimi.ua.ac.be/data/
The pattern (36, 39, 41) occurs in the transactions 572 times (this is also calculated in test), but the FPGrowth returns pattern (36, 39, 41) with support 573.
hace something
Please find attached failing unit test - it's actually a maven project, which contains test data and is ready to run.","org.apache.mahout.fpm.pfpgrowth.fpgrowth.FPGrowth:pruneFPTree(MutableLong, FPTree)"
FILE,AMQP,AMQP-190,2011-09-10T20:24:17.000-05:00,CachingConnectionFactory leaks channels when synchronized with a TransactionManager,"It seems that when I use RabbitTemplate, channelTransacted=true, to convertAndSend() a message to an exchange within the context of a synchronized TransactionManager (e.g. an active transaction on the current thread), the channel is never closed, hence new publishes will always get their ""own"", shiny, new channel (that is never closed or released to the channel pool) until Rabbit can't handle any more channels.
see Forum reference for more info
not observe on consumer side
observe on publishing side
use RabbitTemplate use spring-integration use <int-amqp:outbound-channel-adapter...> tag
supply simple recreate scrounge time","org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer
org.springframework.amqp.rabbit.core.RabbitTemplatePerformanceIntegrationTests
org.springframework.amqp.rabbit.connection.ConnectionFactoryUtils
org.springframework.amqp.rabbit.connection.RabbitResourceHolder"
FILE,AMQP,AMQP-502,2015-06-19T03:02:33.000-05:00,Fanout binding is not created due to missing routing key,"Currently i am using spring-cloud-starter-bus-amqp which in terms references spring-amqp 1.4.3.
when i declare a rabbitlistener like this:
I will get an error, that the binding can not be created.
try exchange type
omit key
The following is the debug logout and the stacktrace:
close connection reopen connection
declare mytest.broadcast
declare 5df237ec-13d4-4aab-a0ff-772707bd7d03
exchange [ mytest.broadcast] with routing route key [ null ]
start on 5df237ec-13d4-4aab-a0ff-772707bd7d03 amq.ctag-Gai8Uo2Q0SYYJDbSopNtLA
raise exception
cause by java.lang.IllegalStateException
close channel on exception
close channel on exception","org.springframework.amqp.rabbit.annotation.EnableRabbitIntegrationTests
org.springframework.amqp.rabbit.annotation.RabbitListenerAnnotationBeanPostProcessor"
FILE,AMQP,AMQP-653,2016-10-08T02:53:08.000-05:00,not take advantage of registered converters,"When using RabbitTemplate in a Spring Boot application, it's very easy to register a Spring AMQP Message Converter.
Just add this to your code:
return new Jackson2JsonMessageConverter()
However, if you switch to RabbitMessagingTemplate, that bean no longer works, because RabbitMessagingTemplate doesn't offer to look up RabbitTemplate's converters, and instead relies on its own.
hook up message converters look inside Spring boot","org.springframework.amqp.rabbit.core.RabbitMessagingTemplateTests
org.springframework.amqp.rabbit.core.RabbitMessagingTemplate"
FILE,AMQP,AMQP-656,2016-10-15T00:25:46.000-05:00,Unable to refer to the default exchange using @Argument within a @RabbitListener,"It seems you are unable to use @Argument annotations that use empty strings to refer to the default exchange.
For example, you should be able to configure a queue to use the default exchange as part of the dead letter config similar to the following:
exchange @Exchange
This fails though as spring seems to not send the empty string.
use things like SPEL evaluate things like SPEL evaluate things to empty string evaluate things to same result
use bean configs get configuration use something like following be with annotation
return QueueBuilder.durable(""app.events.myEvent"")","org.springframework.amqp.rabbit.annotation.EnableRabbitIntegrationTests
org.springframework.amqp.rabbit.annotation.RabbitListenerAnnotationBeanPostProcessor"
METHOD,commons-math-3-3.0,MATH-718,2011-12-03T18:40:44.000-06:00,inverseCumulativeProbability of BinomialDistribution returns wrong value for large trials.,"The inverseCumulativeProbability method of the BinomialDistributionImpl class returns wrong value for large trials.
Following code will be reproduce the problem.
This returns 499525, though it should be 499999.
return infinity
not work as result","org.apache.commons.math3.util.ContinuedFraction:evaluate(double, double, int)"
FILE,DATACMNS,DATACMNS-157,2012-04-20T01:24:38.000-05:00,@Query in extending interface is not picked up correctly,"I try to define an interface method in a super repository interface and 'implement' this in an extending interface with @Query.
Tested in the latest nightly build:
extend EntityRepository<PersonalSiteVehicle> {
join p.detail join d.enrichable
Results in
create bean with carrepository throw exception on object creation not create for method public abstract java.lang.Object nl.inmotiv.indi.repository.EntityRepository.findByDealer(nl.inmotiv.indi.domain.Dealer)
not create for method public abstract java.lang.Object nl.inmotiv.indi.repository.EntityRepository.findByDealer(nl.inmotiv.indi.domain.Dealer)
find for type class nl.inmotiv.indi.domain.PersonalSiteVehicle
It looks like Spring Data does not use the @Query annotation in the sub interface.","org.springframework.data.repository.core.support.DefaultRepositoryInformationUnitTests
org.springframework.data.repository.core.support.DefaultRepositoryInformation"
FILE,DATACMNS,DATACMNS-233,2012-09-14T07:38:12.000-05:00,return null for null sources return null for empty strings,"notice important issue relate to automatic web binding
Imagine the use case where you have an Order domain class which has a ManyToOne reference to Customer.
When posting a new Order where Order.customer == """" then a converter exception is thrown:
convert property value of type java.lang.String convert property value to required type org.mycomp.domain.Customer convert from type java.lang.String convert to type @javax
cause complete blocker for optional references
This is the code I used:","org.springframework.data.repository.support.DomainClassConverterUnitTests
org.springframework.data.repository.support.DomainClassConverter"
FILE,DATACMNS,DATACMNS-257,2012-11-29T02:29:27.000-06:00,not deal with uppercase fields,"Cannot execute MongoOperations.findOne method if my model entity contains @Id field which name is uppercase, like UID.
Here an example:
1) create an entity like in example code snippet in MongoDb
2) try to perform find by id operation by calling MongoOperations.findOne
3) at this step you will get an exception
find on com.xxxxxxxxxxxxx.TemplateDefinitionObject!","org.springframework.data.mapping.PropertyPath
org.springframework.data.mapping.PropertyPathUnitTests"
FILE,DATACMNS,DATACMNS-379,2013-10-08T11:52:55.000-05:00,SortHanderMethodArgumentResolver fails to resolve comma-only sort parameter,"If specified "",""(comma only) in parameter value of sort, occurred the ArrayIndexOutOfBoundsException.
occur IllegalArgumentException occur other patterns","org.springframework.data.web.SortHandlerMethodArgumentResolverUnitTests
org.springframework.data.web.SortHandlerMethodArgumentResolver"
FILE,DATACMNS,DATACMNS-511,2014-05-22T00:04:43.000-05:00,AbstractMappingContext.addPersistentEntity causes infinite loop,"not run tests after updating update from Codd sr2 update to Dijkstra
debug issue find after debugging lie in AbstractMappingContext.addPersistentEntity.
never call method
We use quite a few abstract MappedSuperclasses that have circular references and apparently this does not work.
An example:
extend AbstractPermission<USER> implement TenantEntity<TENANT> {
extend AuditingDateBaseEntity<USER> {
extend AuditingDateBaseEntity<USER> {
implement AuditingEntity<USER> {
implement DateEntity {
implement BaseEntity {",org.springframework.data.util.TypeVariableTypeInformation
FILE,DATACMNS,DATACMNS-562,2014-08-19T01:25:20.000-05:00,resolve TreeMap as map value type,"This is the Class of entity used to save to mongodb.
extend classa {
This is the call entry:
try {
Exception caught as below:
get same issues
have something different return empty Type array to emit ArrayIndexOutOfBoundsException
traverse field
use as container","org.springframework.data.mapping.model.AbstractPersistentPropertyUnitTests
org.springframework.data.mapping.model.AbstractPersistentProperty"
FILE,DATACMNS,DATACMNS-594,2014-11-14T03:39:52.000-06:00,behaviour for nested generic types,"After the resolution of DATACMNS-590, we found another strange behaviour using Spring Data MongoDB, but only disabling the default type mapper resolution, which resolves the class type to map using an attribute.
see github repo run unit tests
use generics in minimal expression possible
take close look
Retrieving a Customer and navigate to ItemType will lead to a ClassCastException due the lost of generic type.
But if we retrieve an Item and navigate to ItemType is ok.
lead into work-as-designed resolution
not use default type not use Spring populate database provide custom type mapper","org.springframework.data.util.ClassTypeInformationUnitTests
org.springframework.data.util.ClassTypeInformation
org.springframework.data.util.ParameterizedTypeInformation
org.springframework.data.util.TypeDiscoverer
org.springframework.data.util.GenericArrayTypeInformation"
FILE,DATACMNS,DATACMNS-616,2014-12-17T02:25:54.000-06:00,AnnotationRevisionMetadata can't access private fields,"Trying to use a custom Envers revision class:
implement Serializable {
triggers this error:
not access member of class ExtendedRevision
make from field callback",org.springframework.data.util.AnnotationDetectionFieldCallback
FILE,DATACMNS,DATACMNS-683,2015-04-13T05:31:25.000-05:00,enable Spring break @ModelAttribute binding in Spring MVC,"Given following Java config class
package be.vdab.web;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.web.config.EnableSpringDataWebSupport;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
enkele imports
extend WebMvcConfigurerAdapter {
, following Controller class
package be.vdab.web;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.servlet.ModelAndView;
import be.vdab.entities.Person;
return new ModelAndView(TOEVOEGEN_VIEW)
throw new illegalargumentexception
return redirect
and following JSP
the method post in PersonController throws the InvalidArgumentException because the person parameter is null.
put @EnableSpringDataWebSupport in comment
put @ModelAttribute in comment
clone project show bug from https://github.com/desmethans/springDataJpaError.git show project from https://github.com/desmethans/springDataJpaError.git","org.springframework.data.repository.support.DomainClassConverterUnitTests
org.springframework.data.repository.support.DomainClassConverter"
CLASS,derby-10.9.1.0,DERBY-5407,2011-09-12T08:50:38.000-05:00,"When run across the network, dblook produces unusable DDL for VARCHAR FOR BIT DATA columns.","In private correspondence, Mani Afschar Yazdi reports that dblook omits the length specification for VARCHAR FOR BIT DATA columns when run across the network.
Embedded dblook runs fine.
I can reproduce this problem as follows:
1 Bring up a server (here I am using port 8246).
2 Create a database with the following ij script:
create table t( a varchar( 20 ) for bit data )
3 Now run dblook across the network:
This produces the following DDL for the table:
A similar experiment using an embedded database produces usable DDL which includes a length specification for the VARCHAR FOR BIT DATA column.","java.testing.org.apache.derbyTesting.functionTests.tests.lang.SystemCatalogTest
java.engine.org.apache.derby.catalog.types.BaseTypeIdImpl"
CLASS,derby-10.9.1.0,DERBY-5638,2012-03-01T13:43:10.000-06:00,intermittent test failure in test_05_ClobNegative when running full largedata._Suite; LobLimitsTestjava.sql.SQLException: Table/View 'BLOBTBL' already exists in Schema 'APP'.,"I've seen the following failure when running the largedata suite:
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
use ms
exist BLOBTBL in APP
exist in APP
find messages
Note, when this happens, the suite exits, so that instead of the expected 20 (or 21 on windows, see DERBY-5624 for reason for skipping on Linux default installs with 1024 max open files) we only get 15 (or 16) tests run - if the test doesn't fail it goes on to run the last 5 fixtures again for network server.","java.testing.org.apache.derbyTesting.junit.CleanDatabaseTestSetup
java.testing.org.apache.derbyTesting.functionTests.tests.largedata.LobLimitsTest
java.testing.org.apache.derbyTesting.functionTests.tests.largedata.LobLimitsLiteTest"
CLASS,derby-10.9.1.0,DERBY-5750,2012-05-08T22:56:42.000-05:00,Sending an empty string as table name to compress table procedure or empty string as index name to update statistics procedure makes the parser throw an exception.,"When empty string is passed for the index name to the update statistics procedure, it throws following exception(passing an empty string for table to compress table procedure also gives simiar exception).
The new stored procedure being created through DERBY-4115 for dropping the statistics also gave similar exception
connect jdbc:derby:db1
create table t1
syntax error evaluate expression throw while evaluating
syntax error evaluate expression throw while evaluating
cause by java.sql.SQLException evaluate expression throw while evaluating
cause by java.sql.SQLException
cause by java.sql.SQLException
issue command for general information issue command on IJ command syntax
treat unrecognized commands as potential SQL commands execute unrecognized commands as potential SQL commands
consult DBMS server reference documentation for details","java.testing.org.apache.derbyTesting.functionTests.tests.lang.UpdateStatisticsTest
java.testing.org.apache.derbyTesting.functionTests.tests.lang.CompressTableTest
java.engine.org.apache.derby.catalog.SystemProcedures"
CLASS,derby-10.9.1.0,DERBY-5838,2012-07-02T07:53:00.000-05:00,"By using syscs_set_database_property, you can set the database version to a garbage string which prevents you from booting the database.","The following script shows this problem:
connect jdbc:derby:db
call syscs_util
-- fails to boot
connect jdbc:derby:db","java.engine.org.apache.derby.impl.sql.LanguageDbPropertySetter
java.testing.org.apache.derbyTesting.functionTests.tests.lang.SQLAuthorizationPropTest"
CLASS,derby-10.9.1.0,DERBY-5911,2012-08-28T06:18:36.000-05:00,WHERE condition getting pushed into sub-query with FETCH,"Derby pushes query conditions down into subqueries with FETCH limits, thus creating wrong results.
Take the following snippet:
fetch next rows
The first query correctly returns the rows (Colombian,5), (Colombian_Decaf,20).
The second query (which filters the result of the first one) returns (Colombian,5), (French_Roast,5).
It shows up because (supposedly) the filter condition has been evaluated before the fetch limit.","java.engine.org.apache.derby.impl.sql.compile.ProjectRestrictNode
java.testing.org.apache.derbyTesting.functionTests.tests.lang.OrderByAndOffsetFetchInSubqueries
java.engine.org.apache.derby.impl.sql.compile.SelectNode"
CLASS,derby-10.9.1.0,DERBY-5916,2012-08-31T23:16:09.000-05:00,java.lang.NullPointerException org.apache.derby.impl.store.raw.data.BaseDataFileFactory.stop() connecting to network server,"I got a report of the exception below trying to connect to  database with absolute path and network server and the database name attribute.
I haven't gotten  information  on the derby version or  platform, Below is the url I received with some characters replaced.
connect jdbc:derby://localhost:1527/;databaseName=/home/uxxxx/installs/hxxx_ext/mxxxxxxxx_db
report on list",java.engine.org.apache.derby.impl.store.raw.data.BaseDataFileFactory
CLASS,derby-10.9.1.0,DERBY-6053,2013-01-25T09:02:53.000-06:00,Client should use a prepared statement rather than regular statement for Connection.setTransactionIsolation,"build up time for setTransactionIsolation() build up Statement for setTransactionIsolation()
The program below shows repeated calls to setTransactionIsolation.
import java.sql.
start own NetworkServer run SQL
reproduce issues
throw Exception {
try {
start Network server
connect to customer database
clean up from previous run
try {
start Network server
throw SQLException {
try {
start retries
try {
sleep second ping network server
not throw exception
start = true
get reply on ping
throw failexception
throw Exception {
throw Exception {",java.client.org.apache.derby.client.am.Connection
METHOD,time,77,2013-10-16T15:36:22.000-05:00,addDays(0) changes value of MutableDateTime,"Upon DST transition from summer to winter time zone, adding the amount of zero days to a mutable date time object changes the value of the object.
The code
date + mdt + ( + mdt.toInstant()
prints
change by hour change start date
The methods addMonths and addYears show the same problem; addSeconds, addMinutes and addHours are ok.
test with version
repeat test with Joda 1.5.2 not change value","org.joda.time.MutableDateTime:add(DurationFieldType, int)
org.joda.time.MutableDateTime:addWeeks(int)
org.joda.time.MutableDateTime:addYears(int)
org.joda.time.MutableDateTime:addMonths(int)
org.joda.time.MutableDateTime:addMinutes(int)
org.joda.time.MutableDateTime:addHours(int)
org.joda.time.MutableDateTime:addWeekyears(int)
org.joda.time.MutableDateTime:addDays(int)
org.joda.time.MutableDateTime:addSeconds(int)
org.joda.time.MutableDateTime:addMillis(int)"
METHOD,time,93,2013-12-01T09:33:58.000-06:00,Partial.with fails with NPE,"With the latest master:
java new Partial(yearOfCentury(), 1)
Fails with yearOfCentury, year and yearOfEra.
have null range duration type","org.joda.time.Partial:Partial(DateTimeFieldType[], int[], Chronology)
org.joda.time.field.UnsupportedDurationField:compareTo(DurationField)"
FILE,COMPRESS,COMPRESS-187,2012-05-23T14:00:58.000-05:00,ZipArchiveInputStream and ZipFile don't produce equals ZipArchiveEntry instances,"I'm trying to use a ZipArchiveEntry coming from ZipArchiveInputStream that I stored somwhere for later with a ZipFile and it does not work.
The reason is that it can't find the ZipArchiveEntry in the ZipFile entries map.
It is exactly the same zip file but both entries are not equals so the Map#get fail.
I looked at ZipArchiveInputStream and it looks like the comment (whatever it is) is simply not parsed while I can find some code related to the comment at the end of ZIipFile#readCentralDirectoryEntry.
not have issue","org.apache.commons.compress.archivers.zip.ZipArchiveEntry
org.apache.commons.compress.archivers.zip.ZipArchiveEntryTest"
FILE,COMPRESS,COMPRESS-189,2012-06-26T21:30:39.000-05:00,ZipArchiveInputStream may read 0 bytes when reading from a nested Zip file,"When the following code is run an error ""Underlying input stream returned zero bytes"" is produced.
If the commented line is uncommented it can be seen that the ZipArchiveInputStream returned 0 bytes.
work with line
The zip file used to produce this behavious is available at http://wwmm.ch.cam.ac.uk/~dl387/test.ZIP
process zip file of zip files
Also I believe whilst ZipFile can iterate over entries fast due to being able to look at the master table whilst ZipArchiveInputStream cannot.
instantiate ZipFile from zip file instantiate ZipFile without extracting extract nested zip file","org.apache.commons.compress.archivers.zip.ZipArchiveInputStreamTest
org.apache.commons.compress.archivers.zip.ZipArchiveInputStream"
FILE,COMPRESS,COMPRESS-252,2013-12-18T16:12:34.000-06:00,Writing 7z empty entries produces incorrect or corrupt archive,"not find exact rule cause incorrect behavior reduce to simple scenarios
Input: A folder with certain files -> tried to archive it.
contain files
Scenario 1: 7 empty files
Result: The created archive contains a single folder entry with the name of the archive (no matter which was the name of the file)
Scenario 2: 7 files, some empty, some with content
Result: The created archive contains a folder entry with the name of the archive and a number of file entries also with the name of the archive.
The number of the entries is equal to the number of non empty files.
Scenario 3: 8 empty files
not open archive stop archive
Scenario 4.1: 8 files: some empty, some with content, last file (alphabetically) with content
describe for Scenario
Scenario 4.2: 8 files, some empty, some with content, last file empy
Result: archive is corrupt, the following message is received: ""Cannot open file 'archivename.7z' as archive"" (7Zip Manager does not crash).","org.apache.commons.compress.archivers.sevenz.SevenZOutputFile
org.apache.commons.compress.archivers.sevenz.SevenZOutputFileTest"
FILE,COMPRESS,COMPRESS-309,2015-02-18T17:22:16.000-06:00,read to full buffer,"BZip2CompressorInputStream.read(buffer, offset, length) returns -1 when given an offset equal to the length of the buffer.
This indicates, not that the buffer was full, but that the stream was finished.
It seems like a pretty stupid thing to do - but I'm getting this when trying to use Kryo serialization (which is probably a bug on their part, too), so it does occur and has negative affects.
Here's a JUnit test that shows the problem specifically:
throw Exception {
create big random piece of data
// Fails, returns -1 (indicating the stream is complete rather than that the buffer
// was full)
change above expected value",org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream
FILE,ENTESB,ENTESB-3099,2015-04-27T11:35:23.000-05:00,build without warnings work without fabric,"Running camel-log following instructions in README gives lots of warnings then fails wilth following error
find for macro
find for macro
not find matching referal for io.fabric8.
not find matching referal for org.apache.camel.
not find matching referal for org.apache.commons.logging.
not find matching referal for org.apache.felix.gogo.commands.
not find matching referal for org.apache.felix.service.command.
not find matching referal for org.osgi.service.component.
not find matching referal for org.springframework.
not find matching referal for org.springframework.osgi.
find matching referal for scala *
install @ beginner-camel-l og
quickstart \ beginner \ camel
log \ target \ shared.jar to c
quickstart \ beginner \ camel
log \ pom.xml to c
quickstart \ beginner \ camel
log \ target \ shared-sources.jar to c
jboss \ quic kstarts \ fuse
quickstart \ beginner \
execute goal io.fabric8:fabric8-maven-plugin:1.2.0
not transfer artifact org.jboss.quickstarts.fuse:beginner-camel-log:pom:6
help ]
see full stack trace of errors
use x switch enable full debug logging
help ] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionE xception",org.jboss.quickstarts.fuse.processor.BodyInAggregatingStrategy
CLASS,bookkeeper-4.1.0,BOOKKEEPER-336,2012-07-09T23:49:01.000-05:00,bookie readEntries is taking more time if the ensemble has failed bookie(s),"1) Start three bookies.
Create ledger with ensemblesize=3, quorumsize=2
2) Add 100 entries to this ledger
3) Make first bookie down and read the entries from 0-99
Output: Each entry is going to fetch from the failed bookie and is waiting for the bookie connection timeout, only after failure going to next bookie.
affect read entry performance
add failed bookie readTimeOut affect namenode switching time","bookkeeper-server.src.main.java.org.apache.bookkeeper.client.RoundRobinDistributionSchedule
bookkeeper-server.src.main.java.org.apache.bookkeeper.client.DistributionSchedule
bookkeeper-server.src.main.java.org.apache.bookkeeper.client.LedgerHandle
bookkeeper-server.src.main.java.org.apache.bookkeeper.conf.ClientConfiguration
bookkeeper-server.src.main.java.org.apache.bookkeeper.client.PendingReadOp
bookkeeper-server.src.main.java.org.apache.bookkeeper.client.LedgerEntry"
FILE,swt-3.1,100040,2005-06-14T14:19:00.000-05:00,Slowdown between 3.1 RC1 and N20050602 due to change to ImageList,"build n20050602
Several of the UI performance tests got slower between RC1 and N20050602, on
Windows.
I tried running one of them, the OpenCloseWindowTest on N20050602, and also with
N20050602+RC1's SWT.
Here are the numbers (on Win2K):
include for debug perspective
The other perspectives showed less of a change, and seemed to correlate with the numer of icons (either in the menus or toolbars or both).
The debug perspective has the most icons.
I'm suspicious of the ImageList change to always specify OS.ILC_MIRROR.
unwind single change make difference","org.eclipse.swt.widgets.TabFolder
org.eclipse.swt.widgets.Button
org.eclipse.swt.widgets.Tree
org.eclipse.swt.widgets.ImageList
org.eclipse.swt.widgets.Display
org.eclipse.swt.widgets.Menu
org.eclipse.swt.widgets.ToolItem
org.eclipse.swt.widgets.Table"
FILE,swt-3.1,104966,2005-07-24T23:17:00.000-05:00,Link displays as enabled when setText called after setEnabled,"run with XP
1. Create a new Visual bean (SWT)
2. Drop a link on the composite
3. Set the enabled property on the link to be disabled
4. Note: the link is now disabled in the preview
close editor
6. Re-open the editor
7. Observe: link now shows as enabled
When you run the application the link shows as enabled, but is actually disabled.
If you change the order of the initialization such that the enabled status is set after the text is set it displays correctly.",org.eclipse.swt.widgets.Link
FILE,swt-3.1,58185,2004-04-12T15:16:00.000-05:00,load oldstyle BMP files,"I have several old style BMP files.
When I attempt to load them into an image eclipse runs out of memory.
WinBMPFileFormat class does not check the header of the file correctly before reading the image.
In my case, I have a BMP file with the following header:
start at offset
tell difference check size","org.eclipse.swt.internal.image.OS2BMPFileFormat
org.eclipse.swt.SWT
org.eclipse.swt.internal.image.FileFormat
org.eclipse.swt.internal.image.WinBMPFileFormat"
FILE,swt-3.1,81264,2004-12-15T13:17:00.000-06:00,Table fails to setTopIndex after new items are added to the table,"keep track into table dynamically keep track of loads content keep track of scroll bar keep table viewer into table dynamically keep table viewer of loads content keep table viewer of scroll bar work on table viewer scroll to end
maintain position of table call setTopIndex at end
create small testcase simulate process
Here's my testcase to demonstrate the problem:
Table.setTopIndex fails to position to the correct table item if new items are added to the table after the shell is opened.
The first call to setTopIndex succeeds.
The table is correctly positioned at item 20.
After adding new table items to the table, calling setTopTable(40) has no effect.
Calling getTopIndex continues to return 20.
expect on windows","org.eclipse.swt.widgets.Tree
org.eclipse.swt.widgets.List
org.eclipse.swt.widgets.Table"
FILE,swt-3.1,83262,2005-01-19T17:41:00.000-06:00,rxvt pastes null terminator,"When pasting using the middle mouse button from StyledText into an rxvt window,
it contains a null terminator at the end.
Pasting from other applications like
gedit or mozilla into rxvt does not show this behaviour, and this is not
reproducable when pasting into an xterm.
exist on Motif GTK +","org.eclipse.swt.dnd.RTFTransfer
org.eclipse.swt.dnd.TextTransfer"
FILE,swt-3.1,84557,2005-02-07T07:06:00.000-06:00,Table item does not appear in table,"run i20050202-0800 linux gtk
ctrl esc E brings up the handy switch to editor window, but unfortuantley the
list is empty :(","org.eclipse.swt.widgets.TableColumn
org.eclipse.swt.widgets.TreeColumn"
FILE,swt-3.1,84609,2005-02-07T13:35:00.000-06:00,TableColumn has NPE while calling pack()  on last column,"Consider followed code, table has only 3 columns:
refresh table on new data lvtTable.getColumn(0)
On third call I get caught NPE (in debugger) in TableColumn (line 356), because parent.getColumns() (in TableColumn:354) returns array with 4 elements (always one more as existing in the table), and the last element is always null.","org.eclipse.swt.widgets.TableColumn
org.eclipse.swt.widgets.Table"
FILE,swt-3.1,86000,2005-02-21T14:47:00.000-06:00,ImageLoader Save - produces invalid JPEG images,"The ImageLoader Save function appears to be producing bad JPG images.
Simple test case below loads
 PNG Files and Saves them as JPEG.
Many files were tested and the majority 
 did produced the proper JPG images as expected.
contain files not save files to JPEG
package com.ibm.test.image;
import org.eclipse.swt.
try { for { string filein = dir + files [ i ] +",org.eclipse.swt.internal.image.JPEGFileFormat
FILE,swt-3.1,87997,2005-03-14T19:21:00.000-06:00,TableEditor.dispose( ) causes NPE if linked Table is being disposed,"find in i20050308-0835
call methods on owning own Table remove listeners from columns
If the table is in the process of being
disposed, the columns will have already been disposed and this will result in a
NPE.
Specifically this prevents one from adding a dispose listener on the Table
or a TableItem and trying to dispose of the associated editor, as in the code
below.
Further if the dispose listener is set on the parent of the Table, a
""Widget is disposed"" exception will be thrown instead of the NPE.
leave place to hook trigger disposal of TableEditor
import org.eclipse.swt.events.","org.eclipse.swt.widgets.Tree
org.eclipse.swt.widgets.Table"
FILE,swt-3.1,88717,2005-03-22T04:34:00.000-06:00,[DND] Tree is scrolling very slow when dragging an Item to the Bottom,"Create a Tree with some items and implement DND support as explained in this
Snippet:
Now expand some items so that vertical scrollbars are appearing.
When draggin an item to the bottom, the Tree is not scrolling.
It is currently not possible to Drag an Item to a position that is out of view.
expand items
drag Bookmark to bottom","org.eclipse.swt.dnd.TreeDragUnderEffect
org.eclipse.swt.dnd.TableDragUnderEffect"
FILE,swt-3.1,93380,2005-05-01T22:12:00.000-05:00,[DnD] 100% CPU usage when dragging files in the navigator or package explorer,"When i drag and drop a file inside the package explorer or navigator views the
CPU says at 100% (or 50% when hyperthreading is enabled).
not go above % spend in event thread
I do not see this massive use of CPU when dragging an external file into the navigator or when dragging a file out of the navigator.
get thread dumps",org.eclipse.swt.dnd.DragSource
FILE,CONFIGURATION,CONFIGURATION-214,2006-05-26T21:35:46.000-05:00,Adding an integer and getting it as a long causes an exception,"Try this in a BeanShell:
not map to Long object org.apache.commons.configuration.ConversionException not map to Long object not map target exception
The problem is that when an object in a property is not a Long, the only attempt of PropertyConverter.toLong() is that of treating it as a string.","org.apache.commons.configuration.TestPropertyConverter
org.apache.commons.configuration.PropertyConverter
org.apache.commons.configuration.TestBaseConfiguration"
FILE,CONFIGURATION,CONFIGURATION-241,2006-12-02T00:03:48.000-06:00,clearProperty() does not generate events,"I am loading configuration information from multiple sources and have registered a listener with the resulting configuration object.
Unfortunately the listener does not receive ""clear property"" events.
I've confirmed that it can properly receive other events (like ""set property""), and that calls to ""clearProperty()"" do actually clear the property, so I believe this may be a bug in commons-configuration.
set details to true have effect have true
Below is a watered down version of what I am doing (note, my configuration file simply pulls in a property file containing this property: name.first=Mike):
get config file","org.apache.commons.configuration.TestCompositeConfiguration
org.apache.commons.configuration.CompositeConfiguration"
FILE,CONFIGURATION,CONFIGURATION-332,2008-07-04T15:54:10.000-05:00,PropertiesConfiguration.save() doesn't persist properties added through a DataConfiguration,"wrap into DataConfiguration
The properties added through a DataConfiguration aren't persisted when the configuration is saved, but they can be queried normally.
not affect commons configuration
The following test fails on the last assertion :
throw ConfigurationException
reload file","org.apache.commons.configuration.TestPropertiesConfiguration
org.apache.commons.configuration.DataConfiguration"
FILE,CONFIGURATION,CONFIGURATION-347,2008-11-05T21:06:22.000-06:00,Iterating over the keys of a file-based configuration can cause a ConcurrentModificationException,"Some implementations of FileConfiguration return an iterator in their getKeys() method that is directly connected to the underlying data store.
When now a reload is performed (which can happen at any time) the data store is modified, and the iterator becomes invalid.
relate ConcurrentModificationExceptions to multi-threading access
But even if the code performing the iteration is the only instance that accesses the configuration, the exception can be thrown.","org.apache.commons.configuration.TestFileConfiguration
org.apache.commons.configuration.AbstractFileConfiguration"
FILE,CONFIGURATION,CONFIGURATION-408,2010-02-11T01:01:05.000-06:00,"When I save a URL as a property value, the forward slashes are getting escaped","When I save a URL as a property value, the forward slashes are getting escaped.
Example Code :
void main(String[] args)",org.apache.commons.configuration.TestPropertiesConfiguration
CLASS,hibernate-3.5.0b2,HHH-4617,2009-11-28T11:42:08.000-06:00,use materialized blobs with Postgresql cause error,"I have entity with byte[] property annotated as @Lob and lazy fetch type, when table is createad the created column is of type oid, but when the column is read in application, the Hibernate reads the OID value instead of bytes under given oid.
write bytea
If i remember well, auto-creating table with Hibernate creates oid column.","org.hibernate.type.CharacterArrayClobType
org.hibernate.type.MaterializedClobType
org.hibernate.type.PrimitiveCharacterArrayClobType
org.hibernate.type.WrappedMaterializedBlobType
org.hibernate.type.MaterializedBlobType
org.hibernate.test.lob.MaterializedBlobTest
org.hibernate.type.BlobType
org.hibernate.type.ClobType
org.hibernate.test.lob.ClobLocatorTest
org.hibernate.dialect.Dialect
org.hibernate.cfg.annotations.SimpleValueBinder
org.hibernate.dialect.PostgreSQLDialect
org.hibernate.Hibernate"
CLASS,hibernate-3.5.0b2,HHH-5042,2010-03-26T05:06:09.000-05:00,TableGenerator does not increment hibernate_sequences.next_hi_value anymore after having exhausted the current lo-range,"introduce new increment variable modify class MultipleHiLoPerTableGenerator.java in version
The problem in the new code is that only value get's incremented whilst variable lo is still used to check when a new hiVal must be obtained.
as lo is never incremented, MultipleHiLoPerTableGenerator continues to deliver numbers without ever update hibernate_sequences.
next_hi_value on the database (only one unique update is propagates at the first insert)
This lead to duplicate keys as soon another session from another sessionfactory tries to insert new objects on the concerning table.
Please see attached testcase.
as the testcase uses 2 sessionfactories hibernate.hbm2ddl.auto=create cannot be used!!
Schema has to be exported separately and the testcase must run without hbm2ddl.auto property!
Here the schema for HSQLDB:
create table
create table hibernate_sequences","org.hibernate.id.SequenceHiLoGenerator
org.hibernate.id.enhanced.OptimizerFactory
org.hibernate.id.SequenceGenerator
org.hibernate.id.MultipleHiLoPerTableGenerator"
METHOD,openjpa-2.0.1,OPENJPA-1627,2010-04-12T05:21:13.000-05:00,ORderBy with @ElementJoinColumn and EmbeddedId uses wrong columns in SQL,"have compound key
The problem is that the order by in the generated SQL is for columns mapped in the transaction entity NOT the TransacionId as expected.
So the Account class has the following fragment....
define _ processDate in TransactionId class define _ tranSequenceNumber in TransactionId class
have following fragment ....
define primary key columns ....
However the generated SQL is doing order by on columns mapped in Transaction:
execute prepstmnt SELECT t0.maccno","org.apache.openjpa.jdbc.meta.JDBCRelatedFieldOrder:order(Select, ClassMapping, Joins)"
METHOD,openjpa-2.0.1,OPENJPA-1762,2010-08-09T14:42:31.000-05:00,javax.persistence.lock.scope EXTENDED doesn't properly lock join tables,"have simple failing unit test exercise small portion of javax.persistence.lock.scope=Extended property exercise simple failing unit test of javax.persistence.lock.scope=Extended property
ManyToMany relationship between A(owner) and B.
Em1 - Find A, no lock
Em1 - Refresh A, PESSIMISTIC_FORCE_INCREMENT and javax.persistence.lock.scope=PessimisticLockScope.EXTENDED -- this SHOULD lock the join table for A_B
Em2 - Find, lock B. Remove B.  should block while trying to remove from Join table.
post unit test demonstrate unit test","org.apache.openjpa.persistence.lockmgr.Employee:toString()
org.apache.openjpa.jdbc.meta.strats.ContainerFieldStrategy:getJoinForeignKey()
org.apache.openjpa.persistence.lockmgr.Employee:getId()
org.apache.openjpa.persistence.lockmgr.Employee:readExternal(ObjectInput)
org.apache.openjpa.persistence.lockmgr.Employee:setDepartment(Department)
org.apache.openjpa.jdbc.kernel.PessimisticLockManager:getLockRows(DBDictionary, Object, ClassMapping, JDBCFetchConfiguration, SQLFactory)
org.apache.openjpa.jdbc.meta.strats.ContainerFieldStrategy:appendSize(SQLBuffer, Select, Joins)
org.apache.openjpa.jdbc.meta.strats.StoreCollectionFieldStrategy:getJoinForeignKey()
org.apache.openjpa.persistence.lockmgr.Employee:writeExternal(ObjectOutput)"
METHOD,openjpa-2.0.1,OPENJPA-526,2008-02-27T13:28:05.000-06:00,Insert text more than 4K bytes to Clob column causes SQLException: Exhausted Resultset,"Here's the persistence class:
public class Exam... {
    @Lob
    @Column(name = ""text"", nullable = false) ***** NOTE: set nullable = true will fix the problem but it leads to bug OPENJPA-525 *****
    private String text;
}
param with nullable = false
end transaction
see nested exceptions for details
not insert NULL
not insert NULL
not insert NULL","org.apache.openjpa.persistence.kernel.common.apps.Lobs:getId()
org.apache.openjpa.persistence.kernel.common.apps.Lobs:getLob()
org.apache.openjpa.persistence.kernel.common.apps.Lobs:Lobs(String, int)
org.apache.openjpa.persistence.kernel.common.apps.Lobs:setLob(String)
org.apache.openjpa.jdbc.sql.OracleDictionary:setNull(PreparedStatement, int, int, Column)"
METHOD,adempiere-3.1.0,1240,2008-05-16T03:03:55.000-05:00,Posting not balanced when is producing more than 1 produc,"The accounting is not balanced  when more that 1 product BOM is produced.
put line
apply in case
1. En Gardenworld , Production windows , create a row in the tab Production Header for  ""Production 2 Patio  set"".
2. in the tab Production Plan create a row for the Patio Furniture Set product, Production Quantity= 2
3. Then click on  ""Create/post Production"" button in the Production header tab, this create the production line.
verify in production line tab
4. the first line is necesary divide in two for give a serial each one.
then in the Patio Furniture Set product set in movement quantity 1 and give a serial  in the attribute set instance field, then create other row similar but the other serial
5. then click on  ""Create/post Production"" button in the Production header tab
6. click on ""Not Postet"" Button, then there the botton label is changed to ""Dont Balanced""",org.compiere.acct.Doc_Production:createFacts(MAcctSchema)
CLASS,pig-0.8.0,PIG-1771,2010-12-16T14:45:37.000-06:00,"New logical plan: Merge schema fail if LoadFunc.getSchema return different schema with ""Load...AS""","The following script fail:
use BinStorage()
dump auxdata
Error message:","test.org.apache.pig.test.TestEvalPipeline2
src.org.apache.pig.newplan.logical.relational.LogicalSchema"
CLASS,pig-0.8.0,PIG-1776,2010-12-17T16:28:09.000-06:00,"changing statement corresponding to alias after explain , then doing dump gives incorrect result","generate group.str
give correct results
/* but dumping c after following steps gives incorrect results */
generate group
generate group.str","src.org.apache.pig.PigServer
src.org.apache.pig.newplan.logical.relational.LOLoad
test.org.apache.pig.test.TestUDFContext"
CLASS,pig-0.8.0,PIG-1813,2011-01-20T10:25:01.000-06:00,Pig 0.8 throws ERROR 1075 while trying to refer a map in the result of  eval udf.Works with 0.7,"generate id generate register myudf.jar; generate id # rmli as rmli:bytearray
The above script fails when run with Pig 0.8 but runs fine with Pig 0.7 or if pig.usenewlogicalplan=false.
The below is the exception thrown in 0.8 :
receive bytearray from UDF convert bytearray to map","src.org.apache.pig.backend.hadoop.executionengine.physicalLayer.expressionOperators.POUserFunc
test.org.apache.pig.test.TestEvalPipeline2
src.org.apache.pig.newplan.logical.relational.LOGenerate"
CLASS,pig-0.8.0,PIG-1837,2011-02-01T17:57:48.000-06:00,use IsEmpty function,"I have the following 2 inputs:
And the following script to work on the data:
grunt data2 generate group grunt by b
After running for a while, I get the following error:
not cast org.apache.pig.builtin.IsEmpty to org.apache.pig.Accumulator","test.org.apache.pig.test.TestAccumulator
src.org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.AccumulatorOptimizer"
CLASS,pig-0.8.0,PIG-1850,2011-02-11T07:50:43.000-06:00,Order by is failing with ClassCastException if schema is undefined for new logical plan in 0.8,"The below is the script :
generate SUM ( $
generate ROUND store e into out1
not cast org.apache.pig.impl.io.NullableDoubleWritable to org.apache.pig.impl.io.NullableBytesWritable
The script is failing while doing order by in WeightedRangePartitioner since it considers the quantiles to be NullableBytesWritable but at run time this is NullableDoubleWritable .
define in load statement
But the same works fine when the  multiquery is turned off.
One more issue worth noting is that if i have a filter statement after relation E, then the above exception is swallowed by Pig.","test.org.apache.pig.test.TestEvalPipeline2
src.org.apache.pig.newplan.logical.relational.LOInnerLoad
src.org.apache.pig.newplan.logical.expression.ProjectExpression"
CLASS,pig-0.8.0,PIG-1854,2011-02-14T03:03:15.000-06:00,Pig returns exit code 0 for the failed Pig script .,"Pig returns exit code 0 for the some of the failed Pig scripts.
like Oozie due_to workflow system show succeeded status instead_of killed status
Suppose in Pig script we have following statement:
On executing the pig script, script failed with following exception
But when executed the similar Pig script through Oozie workflow, Pig script failed but Oozie show the ""succeeded"" status message instead of ""killed"" status as return exit code is 0.
hadoop file system
One more scenario where Pig script failed but exit code is 0 and no exception occurred for this failure.
compile jul
produce result
execute by pig","src.org.apache.pig.impl.util.LogUtils
test.org.apache.pig.test.TestPigRunner
src.org.apache.pig.tools.grunt.GruntParser"
CLASS,pig-0.8.0,PIG-1856,2011-02-15T17:26:16.000-06:00,Custom jar is not packaged with the new job created by LimitAdjuster,"The script:
limit b
be in classpath
The script, however,  fails since the piggybank jar isn't shipped to the backend with the additional job created by the LimitAdjuster.
register piggybank jar in script","src.org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.MRCompiler
test.org.apache.pig.test.TestMRCompiler"
CLASS,pig-0.8.0,PIG-1858,2011-02-17T02:27:48.000-06:00,UDF in nested plan results frontend exception,"The below is my script :
{code}
register myanotherudf.jar;
A = load 'myinput' using PigStorage() as ( date:chararray,bcookie:chararray,count:int,avg:double,pvs:int);
B = foreach A generate (int)(avg / 100.0) * 100   as avg, pvs;
C = group B by ( avg );
D = foreach C {
        Pvs = order B by pvs;
        Const = org.vivek.MyAnotherUDF(Pvs.pvs).
generate Const.sum store d into out_d
fail during compilation
The below is the exception that I get :
try -Dpig.usenewlogicalplan=false.
try -Dpig.usenewlogicalplan=false.
When i trun off new logical plan the script executes successfully.
observe issue",test.org.apache.pig.test.TestEvalPipeline2
CLASS,pig-0.8.0,PIG-1868,2011-02-24T00:42:05.000-06:00,have complex data types from udf,"have complex data types return from eval function
The below is my script :
register myudf.jar; b1 generate tuples generate TransformToMyDataType(Tuples,-1,0,1) as seq generate FLATTEN(seq) generate current.id as id dump c3 foreach by ts
On C3 it fails with below message :
{code}
Couldn't find matching uid -1 for project (Name: Project Type: bytearray Uid: 45 Input: 0 Column: 1)
{code}
The below is the describe on C1 ;
{code}
C1: {seq: {t: (previous: (id: chararray,ts: int,url: chararray),current: (id: chararray,ts: int,url: chararray),next: (id: chararray,ts: int,url: chararray))}}
{code}
The script works if I turn off new logical plan or use Pig 0.7.","src.org.apache.pig.newplan.logical.relational.LogicalSchema
test.org.apache.pig.test.TestSchema"
CLASS,pig-0.8.0,PIG-1893,2011-03-10T20:43:13.000-06:00,Pig report input size -1 for empty input file,"In the following script:
If 1.txt is empty, Pig will report
Successfully read -1 records from: ""1.txt""
have multiinputcounters see in WebUI record from _0_2.txt","src.org.apache.pig.tools.pigstats.JobStats
test.org.apache.pig.test.TestPigRunner"
CLASS,pig-0.8.0,PIG-1911,2011-03-16T16:06:11.000-05:00,Infinite loop with accumulator function in nested foreach,"Sample script:
register v_udf
This causes infinite loop if MyCOUNT implements Accumulator interface.
take function into separate foreach statement take function out_of nested foreach","src.org.apache.pig.backend.hadoop.executionengine.physicalLayer.expressionOperators.POUserFunc
src.org.apache.pig.backend.hadoop.executionengine.physicalLayer.relationalOperators.POForEach
test.org.apache.pig.test.TestAccumulator"
CLASS,pig-0.8.0,PIG-1963,2011-04-04T17:18:24.000-05:00,"in nested foreach, accumutive udf taking input from order-by does not get results in order","This happens only when secondary sort is not being used for the order-by.
For example -
{code}
a1 = load 'fruits.txt' as (f1:int,f2);
a2 = load 'fruits.txt' as (f1:int,f2);
order secondary sort by f2 generate group
explain d","test.org.apache.pig.test.TestAccumulator
src.org.apache.pig.backend.hadoop.executionengine.mapReduceLayer.AccumulatorOptimizer"
CLASS,pig-0.8.0,PIG-1979,2011-04-08T02:24:01.000-05:00,New logical plan failing with ERROR 2229: Couldn't find matching uid -1,"The below is my script 
{code}
register myudf.jar;
c01 = LOAD 'input'  USING org.test.MyTableLoader('');
c02 = FILTER c01  BY result == 'OK'  AND formatted IS NOT NULL  AND formatted !
use PigStorage()
The script is failing in building the plan, while applying for logical optimization rule for AddForEach.
match uid for project
The problem is happening when I try to include doc_005::category in the projection for relation finalresult.
orginate field from udf org.vivek.udfs.MyExtractor
import org.apache.pig.EvalFunc; import org.apache.pig.data.
try { return Schema.generateNestedSchema(DataType.BAG, DataType.CHARARRAY) catch { System.err.println ( error while generating schema
throw new ioexception
The script goes through fine if I disable AddForEach rule by -t AddForEach","test.org.apache.pig.test.TestEvalPipeline2
src.org.apache.pig.newplan.logical.expression.DereferenceExpression"
CLASS,pig-0.8.0,PIG-1993,2011-04-12T19:47:41.000-05:00,PigStorageSchema throw NPE with ColumnPruning,"The following script fail:
store into temp use org.apache.pig.piggybank.storage.PigStorageSchema()
exec temp use org.apache.pig.piggybank.storage.PigStorageSchema()
Error message:","contrib.piggybank.java.src.test.java.org.apache.pig.piggybank.test.TestPigStorageSchema
contrib.piggybank.java.src.main.java.org.apache.pig.piggybank.storage.PigStorageSchema"
CLASS,pig-0.8.0,PIG-730,2009-03-24T14:36:45.000-05:00,"problem combining schema from a union of several LOAD expressions, with a nested bag inside the schema.","use BinStorage as } ) generate flatten(outlinks.target)
---> Would expect both C and D to work, but only C works.
D gives the error shown below.
---> Turns out using outlinks.t.target (instead of outlinks.target) works for D but not for C.","src.org.apache.pig.impl.logicalLayer.schema.Schema
src.org.apache.pig.newplan.logical.expression.DereferenceExpression
src.org.apache.pig.newplan.logical.relational.LogicalSchema
test.org.apache.pig.test.TestSchema"
CLASS,pig-0.8.0,PIG-767,2009-04-15T23:43:29.000-05:00,Schema reported from DESCRIBE and actual schema of inner bags are different.,"The following script:
Prints for the DESCRIBE commands:
The reported schemas for urlContentsG and urlContentsF are wrong.
They are also against the section ""Schemas for Complex Data Types"" in http://wiki.apache.org/pig-data/attachments/FrontPage/attachments/plrm.htm#_Schemas.
sound like technicality
assume inner bag of { chararray } assume UDF of { chararray } not work with { } not work for instance","test.org.apache.pig.test.TestNewPlanLogToPhyTranslationVisitor
src.org.apache.pig.newplan.logical.expression.DereferenceExpression
src.org.apache.pig.newplan.logical.relational.LOInnerLoad
src.org.apache.pig.newplan.logical.rules.DuplicateForEachColumnRewrite
test.org.apache.pig.test.TestLogicalPlanMigrationVisitor
src.org.apache.pig.newplan.logical.relational.LOCogroup
test.org.apache.pig.test.TestSchema
src.org.apache.pig.newplan.logical.relational.LOGenerate"
METHOD,math,MATH-1021,2013-08-10T00:00:22.000-05:00,HypergeometricDistribution.sample suffers from integer overflow,"have application port from commons math
It looks like the HypergeometricDistribution.sample() method doesn't work as well as it used to with large integer values -- the example code below should return a sample between 0 and 50, but usually returns -50.
import org.apache.commons.math3.distribution.HypergeometricDistribution;
trace in debugger do { code } return / (double) getPopulationSize(); {code} it could do: {code} return getSampleSize() * ((double) getNumberOfSuccesses() / getPopulationSize() )",org.apache.commons.math3.distribution.HypergeometricDistribution:getNumericalMean()
METHOD,math,MATH-221,2008-08-29T13:31:56.000-05:00,multiply for complex numbers equal for complex numbers,"relate on complex numbers
little java program + output that shows the bug:
import org.apache.commons.math.complex.
void main(String[] args) {
res:  -0 - 1i
comp: 0 - 1i
res=comp: false
think thats give multiply method",org.apache.commons.math.complex.Complex:equals(Object)
METHOD,math,MATH-358,2010-03-24T17:25:37.000-05:00,ODE integrator goes past specified end of integration range,"handle end as event handle end of integration range
lead in cases lead to error
The following test case shows the end event is not handled properly and an integration that should cover a 60s range in fact covers a 160s range, more than twice the specified range.
throw IntegratorException {","org.apache.commons.math.ode.nonstiff.EmbeddedRungeKuttaIntegrator:integrate(FirstOrderDifferentialEquations, double, double[], double, double[])
org.apache.commons.math.ode.nonstiff.RungeKuttaIntegrator:integrate(FirstOrderDifferentialEquations, double, double[], double, double[])"
METHOD,math,MATH-369,2010-05-03T15:48:27.000-05:00,"BisectionSolver.solve(final UnivariateRealFunction f, double min, double max, double initial) throws NullPointerException","throw NullPointerException as member variable
invoke:
NullPointerException will be thrown.","org.apache.commons.math.analysis.solvers.BisectionSolver:solve(UnivariateRealFunction, double, double, double)"
METHOD,math,MATH-60,2006-05-14T04:20:21.000-05:00,illogical result,"return result from function
po ) of Commons math library
Please see the following code segment for more details:
Note: Similarly, when I passed in the following inputs:
input 2: (source = 1 2 / -3, pos = 0)
input 3: ( source =  -1 -2 / 3, pos = 0)
Function ""Fraction parse(String, ParsePosition)"" returned Fraction 1/3 (means the result Fraction had numerator = 1 and  denominator = 3)for all 3 inputs above.
I think the function does not handle parsing the numberator/ denominator properly incase input string provide invalid numerator/denominator.","org.apache.commons.math.fraction.ProperFractionFormat:parse(String, ParsePosition)"
METHOD,math,MATH-865,2012-09-19T21:10:42.000-05:00,Wide bounds to CMAESOptimizer result in NaN parameters passed to fitness function,"If you give large values as lower/upper bounds (for example -Double.MAX_VALUE as a lower bound), the optimizer can call the fitness function with parameters set to NaN.
divide by infinity",org.apache.commons.math3.optimization.direct.CMAESOptimizer:checkParameters()
METHOD,math,MATH-942,2013-03-09T15:05:04.000-06:00,DiscreteDistribution.sample(int) may throw an exception if first element of singletons of sub-class type,"create array with { { Array.newInstance
An exception will be thrown if:
* {{singleons.get(0)}} is of type T1, an sub-class of T, and
* {{DiscreteDistribution.sample()}} returns an object which is of type T, but not of type T1.
To reproduce:
{code}
List<Pair<Object,Double>> list = new ArrayList<Pair<Object, Double>>();
list.add(new Pair<Object, Double>(new Object() {}, new Double(0)));
list.add(new Pair<Object, Double>(new Object() {}, new Double(1)));
new DiscreteDistribution<Object>(list).
attach patch",org.apache.commons.math3.distribution.DiscreteDistribution:sample(int)
METHOD,math,MATH-949,2013-03-15T18:11:56.000-05:00,LevenbergMarquardtOptimizer reports 0 iterations,"The method LevenbergMarquardtOptimizer.getIterations() does not report the correct number of iterations; It always returns 0.
call BaseOptimizer.incrementEvaluationsCount()
I've put a test case below.
Notice how the evaluations count is correctly incremented, but the iterations count is not.
return new double [ ] { FastMath.pow(point[0], 4) }
return new double [ ] [ ] { {","org.apache.commons.math3.optim.nonlinear.scalar.gradient.NonLinearConjugateGradientOptimizer:doOptimize()
org.apache.commons.math3.optim.nonlinear.scalar.noderiv.CMAESOptimizer:doOptimize()
org.apache.commons.math3.optim.nonlinear.vector.jacobian.GaussNewtonOptimizer:doOptimize()
org.apache.commons.math3.optim.nonlinear.scalar.noderiv.SimplexOptimizer:doOptimize()
org.apache.commons.math3.optim.BaseOptimizer:BaseOptimizer(ConvergenceChecker<PAIR>)
org.apache.commons.math3.optim.nonlinear.scalar.noderiv.PowellOptimizer:doOptimize()
org.apache.commons.math3.optim.nonlinear.vector.jacobian.LevenbergMarquardtOptimizer:doOptimize()"
FILE,WFCORE,WFCORE-58,2014-08-18T05:18:45.000-05:00,Setting some locales causes missing buttons on console and throws errors.,"When WildFly 9 Alpha1 is started with the following system properties in the ""standalone.conf"" then the admin console behaves abnormally and shows some missing buttons in the console.
screenshot mssing_add_remove_disable_Buttons.png
Also it throws following Error while accessing the console URL like [1]","org.jboss.as.domain.management.security.password.PasswordCheckUtil
org.jboss.as.controller.access.rbac.StandardRoleMapper
org.jboss.as.host.controller.discovery.S3Util
org.jboss.as.host.controller.DirectoryGrouping
org.jboss.as.remoting.Protocol
org.jboss.as.test.integration.management.rbac.PermissionsCoverageTestUtil
org.jboss.as.test.integration.management.interfaces.JmxInterfaceStringUtils
org.jboss.as.controller.operations.global.ReadResourceDescriptionHandler
org.jboss.as.threads.ThreadPoolManagementUtils"
FILE,WFCORE,WFCORE-145,2014-10-02T13:28:22.000-05:00,CLI run-batch command cannot process files with comments or blank lines,"In the JBoss CLI tool, the ""run-batch --file=script.cli"" command cannot successfully process lines that start with ""#"" or blank lines.
create cli batch scripts
However, running the batch file containing comments/blank lines as a parameter to jboss-cli.sh works just fine (.
/jboss-cli.sh -c --file=script.cli).
create confusion for users","org.jboss.as.cli.handlers.batch.BatchRunHandler
org.jboss.as.cli.handlers.BaseOperationCommand
org.jboss.as.test.integration.management.cli.BatchFileTestCase"
FILE,WFCORE,WFCORE-267,2014-11-19T19:47:31.000-06:00,CLI prints output twice if using cli client jar,"If you are using the CLI client jar, all output is printed twice.
This is because JBoss logging is not set up and by default CommandContextImpl is printing log messages to standard out.
The output will look something like this:
result =
result =",org.jboss.as.cli.CommandLineMain
FILE,WFCORE,WFCORE-495,2015-01-12T08:48:29.000-06:00,"WFLY won't startup due to ""WFLYCTL0212: Duplicate resource [(\""deployment\"" => \""xxx.war\"")]""","WFLY won't startup due to ""WFLYCTL0212: Duplicate resource [(\""deployment\"" => \""xxx.war\"")]""
If you firstly deploy a .
war application by CLI command, then deploy 2nd time same .
war application by copying it to /depolyment directory.
After server restart, it shows:
fail in unrecoverable manner exiting
see previous messages for details
1. start wildfly and deploy a .
war application by CLI.
2. copy same .
war application to /deployment directory
3. restart wildfly to see the error message.
This happens because step 2 does a ""full-replace-deployment"" operation which does not remove content from standalone/data/content/aa/2d0425dd53572294d591b56efdee2680539eaf/content and deployment info from configuration file(standalone.xml).
Therefore, you will have xxx.war in standalone/data/content and configuration file(standalone.xml), also a xxx.war and xxx.war.deployed file inside /standalone/deployments.
A second time server restart will cause a duplicate resource error.",org.jboss.as.server.deployment.DeploymentFullReplaceHandler
FILE,WFCORE,WFCORE-442,2014-12-02T19:15:13.000-06:00,AbstractMultiTargetHandler-based handlers do not propagate failures to the top level failure-description,"An example is worth a thousand words:
result =
address = > [ ]
result =
name = > jboss.server.temp.dir
address = > [ ]
result =
name = > user.home
address = > [ ]
result =
name = > jboss.server.base.dir
address = > [ ]
result =
name = > java.home
address = > [ ]
result =
name = > user.dir
address = > [ ]
result =
name = > jboss.server.data.dir
address = > [ ]
result =
name = > jboss.home.dir
address = > [ ]
result =
name = > jboss.server.log.dir
address = > [ ]
result =
name = > jboss.server.config.dir
address = > [ ]
result =
name = > jboss.controller.temp.dir
One item in the set has a failure description but the overall response does not.
handle similar things have logic for creating create overall failure-description",org.jboss.as.controller.operations.global.GlobalOperationHandlers
FILE,WFCORE,WFCORE-763,2015-06-16T14:59:39.000-05:00,"Persisting an empty list [] gets read as a list with empty string [""""] resulting in IllegalArgumentException","not handle empty lists in symmetric fashion
e.g. if I persist an empty list to xml, and read it back again, using the default parser/marshaller, I should get back an empty list.
Instead I get back a list of size 1 containing an empty string.
infinispan aliases infinispan modcluster proxies
write empy list
have list of size contain empty string
not allow empty name segment for infinispan","org.jboss.as.controller.AttributeParser
org.jboss.as.controller.SimpleListAttributeDefinitionUnitTestCase"
FILE,WFCORE,WFCORE-754,2015-06-02T11:17:26.000-05:00,start with bmanagement start with specific IP settings,"EAP fails to start with -bmanagement= option.
IPv4 address is used in this option.
Host Controller starts to listening on IPv4:9999
Server:server-one probably resolve IPv4 address to host name.
Server-one tries to connect to this host name.
Hostname is resolved by ping ($ ping dev98) to IPv6 address, where host controller is not listening.
resolve hostname to ipv4 address resolve hostname on machine
not occur on EAP 6.4.0
Example on dev98 machine:
ping dev98 [ 10.16.92.7 ] with bytes
Actual results:
start service jboss.server-boot-operations",org.jboss.as.host.controller.ServerInventoryService
FILE,WFCORE,WFCORE-815,2015-07-13T07:57:45.000-05:00,One profile can have more ancestors with same submodules,"One profile can have more ancestors with same submodules.
It leads to WFLYCTL0212: Duplicate resource [(""subsystem"" => ""subsystem_name"")] .
add hierarchical composition of profiles add hierarchical composition to AS
get fresh EAP
define wflyctl0401 in mail-02
not support } not support overriding subsystems
add subsystem to default-new profile","org.jboss.as.domain.controller.operations.ProfileIncludesHandlerTestCase
org.jboss.as.domain.controller.operations.SocketBindingGroupIncludesHandlerTestCase
org.jboss.as.host.controller.logging.HostControllerLogger"
FILE,WFCORE,WFCORE-798,2015-07-07T19:42:47.000-05:00,miss for EAP version,"Trying to run WildFly full's ParseAndMarshalModelsTestCase against the EAP 6.3 configs results in failures like this:
have attribute isolate to schema parsing",org.jboss.as.domain.management.parsing.ManagementXml_Legacy
FILE,WFCORE,WFCORE-896,2015-08-20T11:24:44.000-05:00,not patch EAP exclude Docs from installation,"EAP installer allows user to exclude some packs from installation.
Docs directory is one of them.
If user choose to do so, $JBOSS_HOME/docs directory is not present in distribution.
Trying to apply a patch (contains changes with docs files) on such a distribution gives me:
block customers use installer from consuming use customers from consuming get EAP consume patches for CP releases
Patch was generated using:","org.jboss.as.patching.metadata.PatchXml
org.jboss.as.patching.runner.ModuleRemoveTask
org.jboss.as.patching.runner.ModuleUpdateTask
org.jboss.as.patching.tests.PatchStepAssertions
org.jboss.as.patching.runner.BundlePatchingTask
org.jboss.as.patching.runner.FileUpdateTask
org.jboss.as.patching.runner.IdentityPatchRunner
org.jboss.as.patching.tests.AbstractPatchTestBuilder
org.jboss.as.patching.runner.FileRemoveTask
org.jboss.as.patching.runner.PatchingTaskDescription
org.jboss.as.patching.metadata.PatchXmlUtils
org.jboss.as.patching.runner.AbstractPatchingTask
org.jboss.as.patching.metadata.ContentModification
org.jboss.as.patching.runner.PatchingTask
org.jboss.as.patching.runner.ContentModificationUtils
org.jboss.as.patching.metadata.RollbackPatchXml_1_0
org.jboss.as.patching.logging.PatchLogger
org.jboss.as.patching.metadata.PatchXml_1_0
org.jboss.as.patching.metadata.ModificationBuilderTarget"
FILE,WFCORE,WFCORE-955,2015-08-27T14:34:07.000-05:00,Server is not responding after attempt to set parent of profile to non-existent profile,"Server is not responding after attempt to set parent of profile to non-existent profile.
Server is not responding also after attempt to set parent of socket-binding-group to non-existent socket-binding-group.
work on wildfly-core ( 2.0.0
occur on wildfly occur on EAP 7.0.0
Get fresh EAP
Actual results:
Get fresh EAP","org.jboss.as.controller.OperationContextImpl
org.jboss.as.controller.AbstractOperationContext
org.jboss.as.controller.logging.ControllerLogger
org.jboss.as.controller.SocketCapabilityResolutionUnitTestCase
org.jboss.as.controller.capability.registry.IncludingResourceCapabilityScope
org.jboss.as.controller.AbstractCapabilityResolutionTestCase"
FILE,WFCORE,WFCORE-868,2015-08-05T08:12:55.000-05:00,CLI if/else is not handling special characters strings with parenthesis,"When using CLI if-else scripting that uses a string that contains parenthesis it fails with the following message:
unexpect operation in ( outcome = = success unexpect operation in result.value","org.jboss.as.cli.handlers.ifelse.LesserThanOperation
org.jboss.as.cli.parsing.ifelse.test.IfExpressionParsingTestCase
org.jboss.as.cli.handlers.ifelse.BaseOperation
org.jboss.as.cli.handlers.ifelse.EqualsOperation
org.jboss.as.cli.handlers.ifelse.IfElseControlFlow
org.jboss.as.cli.handlers.ifelse.AndOperation
org.jboss.as.cli.impl.ArgumentWithValue
org.jboss.as.cli.parsing.DefaultStateWithEndCharacter
org.jboss.as.cli.handlers.ifelse.IfHandler
org.jboss.as.cli.parsing.ifelse.test.ComparisonTestBase
org.jboss.as.cli.handlers.ifelse.OrOperation
org.jboss.as.cli.handlers.ifelse.NotLesserThanOperation
org.jboss.as.cli.handlers.ifelse.NotGreaterThanOperation
org.jboss.as.cli.handlers.ifelse.GreaterThanOperation
org.jboss.as.cli.handlers.ifelse.NotEqualsOperation"
FILE,WFCORE,WFCORE-940,2015-08-31T12:20:01.000-05:00,fool CLI,"confuse by early connection confuse to domain
CLI acts as I was connected to the standalone instance, some high level commands are not executable.
# run while domain is booting
# domain has fully booted
not allow host in standalone mode",org.jboss.as.cli.impl.CommandContextImpl
FILE,WFCORE,WFCORE-931,2015-08-28T12:00:07.000-05:00,Embedded host-controller throws NPE on regular shutdown,"Invoking the /host=master:shutdown operation on embedded host-controller gives me NPE.
name shutdown at address [ ]
Using operation style :shutdown throws NPE.
tab completion for operation works
execute operation.
close resource org.jboss.as.cli.embedded.ThreadContextsModelControllerClient@626966ee",org.jboss.as.host.controller.model.host.HostResourceDefinition
FILE,WFCORE,WFCORE-1007,2015-09-24T06:45:11.000-05:00,Warnings about missing notification descriptions when an operation removes an extension,"When I use migration operation the console log is filled with warning messages of type
not describe notification of type resource-removed not describe notification for resource
If I do the sequence of operation
then I the log looks like
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource
not describe notification of type resource-removed not describe notification for resource","org.jboss.as.controller.AbstractOperationContext
org.jboss.as.controller.logging.ControllerLogger"
FILE,WFCORE,WFCORE-1027,2015-10-01T18:16:10.000-05:00,scop roles,"Setting up host scoped roles as follows https://gist.github.com/heiko-braun/0dc810ed04db8739defd there are inconsistent results in the filtering.
When using a role which only selects the master there is no access-control response header showing the filtered resources, and the slave wrongly appears in the results:
result =
address = > [ ]
result =
name = > master
interface =
address = > [ ]
When using a role that only selects the slave we get a proper access-control header
result = >
address = > [ ]
result = >
address = > [ ]
result =
name = > slave
interface =","org.jboss.as.test.integration.domain.rbac.RBACProviderHostScopedRolesTestCase
org.jboss.as.test.integration.domain.rbac.AbstractHostScopedRolesTestCase
org.jboss.as.controller.operations.global.GlobalOperationHandlers
org.jboss.as.test.integration.domain.rbac.JmxRBACProviderHostScopedRolesTestCase
org.jboss.as.test.integration.domain.rbac.ListRoleNamesTestCase
org.jboss.as.test.integration.domain.rbac.WildcardReadsTestCase"
FILE,WFCORE,WFCORE-997,2015-09-22T03:53:31.000-05:00,"Security realm using ldaps hangs forever during SSL handshake, when ldap server is killed","hit during failover testing
When ldap server is killed in particular time of ssl handshake EAP hangs and waits forever on response, which will never come.
Causing thread to block forever.
use ldaps without specifying specify com.sun.jndi.ldap.connect.timeout value see same problem in LdapLoginModule
add option declare com.sun.jndi.ldap.connect.timeout for security realm provide default non-empty value","org.jboss.as.domain.management.connections.ldap.ThreadLocalSSLSocketFactory
org.jboss.as.domain.management.security.WrapperSSLContext"
FILE,WFCORE,WFCORE-1062,2015-10-20T09:33:46.000-05:00,Deployments from the domain/data/content folder goes missing after few minutes if the Host controller is started with --backup option.,"Host controller started with --backup option deletes the deployments from the domain/data/content folder after approximately 10 mins.
The deployment is restored or recreated if the HostController is started.
1. Setup domaincontroller
2. Setup hostcontroller
3. Start domaincontroller
4. Start hostcontroller with --backup option.
5. deploy an application using cli or admin console
6. deployments are available under $JBOSS_HOME/domain/data/content
7. Wait for approximately 10 minutes
8. Cached deployments are deleted on hostcontroller (domain/data/content is empty).
Following is the logging seen on the console :
remove from location /NotBackedUp/JAR-folder/EAPs/eap6/jboss-eap-6.4.3-Patched/domain/data/contenta27fa1b82e6a92fb56ee25bc76f2625249 / content
9. After restarting the Host Controller, the deployment is restored and the cached deployment can now be seen under the Host Controller's domain/data/content folder.
Actual results: Deployment folder form domain/data/content folder gets deleted after approx 10 mins and are restored if the Host Controller is restarted
not affect availability of application","org.jboss.as.domain.controller.resources.DomainDeploymentResourceDefinition
org.jboss.as.controller.persistence.FilePersistenceUtils
org.jboss.as.repository.ContentRepository
org.jboss.as.domain.controller.operations.deployment.DeploymentAddHandler
org.jboss.as.server.deployment.ContentCleanerService"
FILE,WFCORE,WFCORE-1214,2015-12-11T23:17:45.000-06:00,Operation headers not propagated to domain servers when 'composite' op is used,"When the user adds request headers to an op, they are not propagated to the servers during domain rollout if the 'composite' op is involved.
For example, if I add some stdout printing of what the headers are on the various processes and invoke this:
Then on a HC with two servers, this is logged:
add at location /Users/bstansberry/dev/wildfly/wildfly-core/dist/target/wildfly-core-2.0.5
add at location /Users/bstansberry/dev/wildfly/wildfly-core/dist/target/wildfly-core-2.0.5
Note the CLI 'deploy' is far from the only time the 'composite' op is used.
Among other places, the high level CLI 'batch' command in a domain involves use of 'composite'.","org.jboss.as.domain.controller.operations.coordination.DomainRolloutStepHandler
org.jboss.as.domain.controller.operations.coordination.OperationCoordinatorStepHandler"
FILE,WFCORE,WFCORE-1198,2015-12-09T09:30:00.000-06:00,CLI does not resolve multiple properties if one property is undefined,"Multiple property substitution is working with EAP 6.4.3+, however, if a variable amongst the multiple variables is empty or has no value, then the subsequent property in the CLI command is not substituted.
For example :
and if I execute "".
have following in host.xml
Note APP-VERSION had no value, and so the subsequent SERVER-INSTANCE-NUMBER was not properly resolved","org.jboss.as.cli.parsing.test.PropertyReplacementTestCase
org.jboss.as.cli.parsing.ExpressionBaseState"
FILE,WFCORE,WFCORE-1288,2016-01-11T18:27:07.000-06:00,installing CP over one-off fails if the modules patched by the one-off are not patched in the CP,"If a CP is installed over active one-off patches, the patch application may fail if the CP does not contain updates for the modules that were patched by the one-offs.
accumulate one-off patches release since previous CP not discover bug for long time","org.jboss.as.patching.installation.InstallationManager
org.jboss.as.patching.installation.MutableTargetImpl
org.jboss.as.patching.runner.PatchingTask
org.jboss.as.patching.runner.IdentityPatchRunner
org.jboss.as.patching.runner.IdentityPatchContext"
FILE,WFCORE,WFCORE-1229,2015-12-15T22:24:13.000-06:00,Reloading the embed host controller logs ERROR messages,"Embed host controller sometime logs ERROR message while booting up after the reload ('reload --host=master --admin-only=true').
fail ] ) register resource definition for address [ ] register ] ) for address [ ]
see error on local machine try bare-metal servers in MWQE lab try bare-metal servers in error poped-up
run with 7.0.0
appear in extreme cases appear after other reload operation",org.jboss.as.domain.controller.operations.coordination.PrepareStepHandler
FILE,WFCORE,WFCORE-701,2015-05-19T15:06:17.000-05:00,report between server-config resource report between server resource,"When a managed server fails in some way, the server status reporting is inconsistent between the /host=<host>/server-config=<server> resources and the /host=<host>/server=<server> resource.
To reproduce, run domain.sh, find the pid of a server process, and kill -9 <thepid>.
Then with the CLI:
result = > FAILED
result = > STOPPED",org.jboss.as.host.controller.ManagedServer
FILE,WFCORE,WFCORE-1570,2016-05-27T12:51:56.000-05:00,save name id attribute discrepancy,"When using rollout plans for EAP deployment scenarios I can create my own named rollout-plan for ease of use.
I can then apply rollout command later on, referring with name of my own rollout plan that should be used.
create such rollout plan create way use such rollout plan use way
When I create rollout-plan, I use command like:
see --name attribute given to name my rollout plan
When I then refer to it I use following command:
see id attribute given to rollout header operation
use examples from documentation
miss something retrieve more info use rollout header operation in CLI use rollout header operation in deploy command","org.jboss.as.cli.parsing.operation.header.RolloutPlanState
org.jboss.as.cli.parsing.operation.header.RolloutPlanHeaderCallbackHandler
org.jboss.as.cli.operation.impl.RolloutPlanCompleter"
FILE,WFCORE,WFCORE-1611,2016-06-22T08:49:33.000-05:00,CLI could stuck if SIGINT is send during authentication dialogue,"Interruption a CLI session with Ctrl+C during authentication dialogue could get CLI into unresponsive state.
Either use a remote server, --no-local-auth cli script arg or remove a local element from ManagementRealm configuration
start a standalone server and launch a CLI
Username: <Press Ctrl+C here>
Could not connect message is printed, but process is not terminated and stuck at this point.
not connect to http-remoting
not connect to http-remoting
recover terminal be to SIGKILL",org.jboss.as.cli.impl.CliShutdownHook
FILE,WFCORE,WFCORE-1578,2016-06-07T05:13:13.000-05:00,add local | remote-destination-outbound-socket-binding,"Lets have some /socket-binding-group=standard-sockets/socket-binding with particular name.
Then when I create some /socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding or /socket-binding-group=standard-sockets/local-destination-outbound-socket-binding using same name as of already existing socket-binding resource, add operation is successful but when I perform server reload, it crashes as it is not able to parse configuration.
Start EAP and log to CLI create your own socket-binding resource and {remote|local}-destination-outbound-socket-binding resource with same names and perform reload
server crashes with following stacktrace in console log:
bind INFO [ org.wildfly.extension.undertow] wflyut0007 to 127.0.0.1:8080
declare message in socket-binding-group standard-sockets
fail in unrecoverable manner exiting
see previous messages for details
remove duplicate resources manually by removing
please change","org.jboss.as.controller.logging.ControllerLogger
org.jboss.as.server.services.net.LocalDestinationOutboundSocketBindingAddHandler
org.jboss.as.server.services.net.SocketBindingAddHandler
org.jboss.as.server.services.net.RemoteDestinationOutboundSocketBindingAddHandler"
FILE,WFCORE,WFCORE-1635,2016-07-05T07:04:51.000-05:00,Write attribute on a new deployment scanner fails in batch,"Creating a new deployment-scanner and altering it's attribute fails if done in single batch.
Running the commands without batch or running batch on CLI embed-server works fine.
use embed server",org.jboss.as.server.deployment.scanner.AbstractWriteAttributeHandler
FILE,WFCORE,WFCORE-1590,2016-06-12T14:18:43.000-05:00,Default parameter length validating ignores setMinSize(0),"With the following attribute definition: -
The following error is reported if an empty string is used as a parameter: -
have minimum length of characters","org.jboss.as.controller.operations.validation.BytesValidator
org.jboss.as.controller.SimpleAttributeDefinitionUnitTestCase
org.jboss.as.controller.test.WriteAttributeOperationTestCase
org.jboss.as.controller.AbstractAttributeDefinitionBuilder
org.jboss.as.controller.AttributeDefinition"
FILE,WFCORE,WFCORE-1657,2016-07-15T11:51:17.000-05:00,add module,"Using tab completion for adding a module dependencies with module add command could lead to invalid module.xml as CLI prompt user to use module name with ""system.layers.base"" prefix, which is not a part of module name.
Add a new module via module add command, use tab-completion for dependencies
org sun
this way something like following is generated in module.xml file",org.jboss.as.cli.handlers.module.ASModuleHandler
FILE,WFCORE,WFCORE-1793,2016-09-14T08:08:21.000-05:00,add-content operation fails to overwrite existing content with overwrite=true set when passing content by file path,"Upon overwriting content in managed exploded deployments on wildfly-core, the following errors are produced:
update content of exploded deployment
update content of exploded deployment
update content of exploded deployment
pass content to server","org.jboss.as.server.controller.resources.DeploymentAttributes
org.jboss.as.server.deployment.ExplodedDeploymentAddContentHandler"
FILE,WFCORE,WFCORE-1908,2016-10-31T08:13:57.000-05:00,Tab completion suggest writing attribute which has access type metric and is not writable,"CLI tab completion suggests attributes that are not writable and their access-type is metric
Example
execute read-resource-description be of type metric be from executing
On attempt to write metric attribute, for example message-count, non writable error is printed","org.jboss.as.cli.impl.AttributeNamePathCompleter
org.jboss.as.cli.parsing.test.AttributeNamePathCompletionTestCase
org.jboss.as.cli.Util"
FILE,WFCORE,WFCORE-1936,2016-11-04T10:57:06.000-05:00,not match reality for socket-binding not match reality for *,"If you tries to change such attributes you are informed that reload is necessary.
define attributes as restart-required = > no-services","org.jboss.as.server.services.net.OutboundSocketBindingResourceDefinition
org.jboss.as.controller.resource.AbstractSocketBindingResourceDefinition"
FILE,WFCORE,WFCORE-1959,2016-11-08T16:28:30.000-06:00,Deploying an empty managed exploded deployment to server group in domain fails,"Deploying an empty exploded deployment created on domain controller fails with the following:
roll back operation on servers",org.jboss.as.domain.controller.operations.coordination.ServerOperationResolver
METHOD,tika-1.3,TIKA-1109,2013-04-18T06:06:06.000-05:00,Metadata not extracted before the content in OOXML (pptx),"It seems that when processing OOXML documents, the metadata is only read after the text.
use medata while processing process text
As a symptom:
outputs only as metadata:",org.apache.tika.parser.microsoft.ooxml.XSSFExcelExtractorDecorator:getMetadataExtractor()
CLASS,jedit-4.3,1520774,2006-07-11T20:43:16.000-05:00,enter field in VFSFileChooser File,"If I type '/' + enter in the File-Browser I get the
file list in the attatched screen shot.
be from activity log
vf false \ [ debug \ ] vfsfilenamefield debug \ ] workthread run = roots run path2 run in work thread run in work thread
In 4.2 Final, just typeing '/' took me to the root
directory ""/"".",org.gjt.sp.jedit.MiscUtilities
CLASS,jedit-4.3,1536064,2006-08-07T15:14:23.000-05:00,focus in filesystembrowser NOT in filenamefield,"When I try to open a file \(Ctrl-O\) the focus in the filesystembrowser is on the Commands menu instead of the filename field.
be because GUIUtilities.requestFocus\(\)
not do job
see rev.5386 be to method
prompt by bug-report http://sourceforge.net/tracker/ index.php?func=detail&aid=1275607&group\_id=588&atid=300588
trigger bug indicate bug not need particular patch for bug resolution
revert patch resolve by reverting
not have idea about reason",org.gjt.sp.jedit.GUIUtilities
CLASS,jedit-4.3,1542026,2006-08-17T14:41:54.000-05:00,Keyboard navigation through menubar inserts char into buffer,"Using the keyboard \(characters, or space for 
checkboxes \[return for checkboxes works correctly\]\) 
to navigate through the menubar inserts the pressed 
characters literally in the current buffer if the 
choice is not a sub-/menu, but an action, even if the 
pressed character is not anything choosable \(e. g. 
Alt+p -&gt; a\)
have many bugs check for dupes",org.gjt.sp.jedit.jEdit.MyFocusManager
CLASS,jedit-4.3,1571752,2006-10-05T21:26:12.000-05:00,fold wrong comments in PHP mode,"Before 'Add Explicit fold' the content of buffer looks like this \('X' means selection boundaries\):
After:
function foo \
be between &lt; function empty line work OK",org.gjt.sp.jedit.textarea.TextArea
CLASS,jedit-4.3,1593359,2006-11-09T12:26:59.000-06:00,JEditTextArea processes keyboard input that is not for it,"JEditTextArea processes keyboard input that is not meant for it.
e. g. if focus is on Activity Log
Dockable or Java Insight Dockable, pressing Del-key deletes characters from the buffer.
have many bugs check for dupes",org.gjt.sp.jedit.gui.InputHandler
CLASS,jedit-4.3,1724940,2007-05-24T15:02:18.000-05:00,type in multiple,"If I highlight multiple selections of text in the text area and then begin typing, only the first character of what I type is inserted in the selected areas \(except for where the cursor ended up after making the selection\).
For example if I have the text:
and I highlight both p's in the &lt;p&gt; tags and then type ""div"" I end up with:
attach screenshot",org.gjt.sp.jedit.textarea.BufferHandler
CLASS,jedit-4.3,1739675,2007-06-19T11:44:12.000-05:00,error in jedit svn revision,"After compile and update my jedit to pre10 \(revision 9820+\) i got some erros trying to use c+i with python files.
attach file see error",org.gjt.sp.jedit.indent.WhitespaceRule
CLASS,jedit-4.3,1768638,2008-01-14T19:10:55.000-06:00,check after buffer switch,"implement patch in 4.3pre12
concern general option visit buffer save buffer
switch buffers with RecentBufferSwitcher plug-in
1\.
Open a clean jEdit session, i.e. rename your .
jedit settings directory to something else
2\.
Start jEdit
3\.
Open Global Options and select General
4\.
Set option 'if open files are changed on disk' to 'prompt'
5\.
Set option 'Check for file change upon' to 'visiting or saving the buffer'
6\.
Create two buffers and save them, e.g. f1 and f2
7\.
Select buffer f1
8\.
Overwrite buffer f2 from the command line, e.g. echo Hello &gt; f2
9\.
Switch back to jEdit
10\.
Switch to buffer f2 with the buffer switcher
Nothing happens
Switch focus to another window and back to jEdit again.
You're prompted to take an action for buffer f2
Install the RecentBufferSwitcher
15\.
Open the Global options and select a shortcut for Open Buffers command, e.g. C+TAB.
16\.
Select the f1 buffer again
17\.
Overwrite the f2 buffer from command line again
18\.
Switch back to jEdit
19\.
Switch buffer with C+TAB
You're immediately prompted to take an action for the modified f2 buffer.","org.gjt.sp.jedit.View
org.gjt.sp.jedit.search.SearchAndReplace
org.gjt.sp.jedit.options.GeneralOptionPane
org.gjt.sp.jedit.Buffer
org.gjt.sp.jedit.EditServer
org.gjt.sp.jedit.jEdit
org.gjt.sp.jedit.View.WindowHandler
org.gjt.sp.jedit.gui.CloseDialog.ActionHandler"
CLASS,jedit-4.3,1942313,2008-04-14T18:08:29.000-05:00,Expanding a fold twice moves caret unexpectedly,"use 4.3pre13 on Debian testing \ use 4.3pre13 on Ubuntu feisty \
get same results on machines
Create the following file, and save it as test.cpp:
void bcd \
void efg \
void hij \
Change the folding mode \(if necessary\) to \`indent'.
Move the caret to the \`hij' line.
Press Alt+Backspace to contract the fold.
Press Alt+Enter to expand the fold.
Press Alt+Enter a second time.
The caret moves unexpectedly to the \`bcd' line.",org.gjt.sp.jedit.textarea.DisplayManager
CLASS,jedit-4.3,1999448,2008-08-23T10:28:24.000-05:00,Unnecesarry fold expantion when folded lines are edited,"test patch
avoid serious black hole bugs apply patch
\(Quoted from Matthieu's comment for patch \#1999448\)
if I use explicit fold, with this buffer
all folds are folded.
I remove one ""l"" from hello.","org.gjt.sp.jedit.textarea.BufferHandler
org.gjt.sp.jedit.textarea.DisplayManager
org.gjt.sp.jedit.textarea.TextArea"
CLASS,jedit-4.3,2017514,2008-07-13T20:48:23.000-05:00,context menu,"open new bug
Summary: Select all \(Ctrl+A\), then right-click in the selection \(as if to copy, or cut\).
About one in 3 or 4 times, I lose the selection.",org.gjt.sp.jedit.textarea.MouseHandler
CLASS,jedit-4.3,2129419,2008-09-25T23:53:11.000-05:00,NPE in EditPane.setBuffer when quitting jEdit,"When trying to quit jEdit, I get the following Null-Pointer-Exception, which is probably related to some files being changed/deleted \(due to a ""cvs up"" in the background\).
quit jEdit
dialog already failed, so that ""Close"" was the only option that worked.
The NPE:",org.gjt.sp.jedit.gui.CloseDialog.ListHandler
CLASS,jedit-4.3,2824045,2009-07-20T00:56:45.000-05:00,Toggle FullScreen Mode: Window ends up under others,"Sometimes when I toggle fullscreen mode, the jEdit window appears behind other windows \(such as my desktop panel\).
This happens in BOTH directions \(switching it on and off\).
have focus toggle between jedit toggle between window",org.gjt.sp.jedit.View
